diff --git a/config/cc/gcc.in b/config/cc/gcc.in
index e67e435..f9504c7 100644
--- a/config/cc/gcc.in
+++ b/config/cc/gcc.in
@@ -44,7 +44,7 @@ config CC_GCC_V_5_2_0
 
 config CC_GCC_V_linaro_4_9
     bool
-    prompt "linaro-4.9-2015.06"
+    prompt "linaro-4.9-2017.01"
     depends on CC_GCC_SHOW_LINARO
     select CC_GCC_4_9
 
@@ -368,7 +368,7 @@ config CC_GCC_VERSION
 # Don't remove next line
 # CT_INSERT_VERSION_STRING_BELOW
     default "5.2.0" if CC_GCC_V_5_2_0
-    default "linaro-4.9-2015.06" if CC_GCC_V_linaro_4_9
+    default "linaro-4.9-2017.01" if CC_GCC_V_linaro_4_9
     default "4.9.3" if CC_GCC_V_4_9_3
     default "linaro-4.8-2015.06" if CC_GCC_V_linaro_4_8
     default "4.8.5" if CC_GCC_V_4_8_5
diff --git a/scripts/build/cc/100-gcc.sh b/scripts/build/cc/100-gcc.sh
index 5674c30..ba406e9 100644
--- a/scripts/build/cc/100-gcc.sh
+++ b/scripts/build/cc/100-gcc.sh
@@ -29,7 +29,7 @@ do_gcc_get() {
             YYMM=`echo ${CT_CC_GCC_VERSION} |cut -d- -f3 |${sed} -e 's,^..,,'`
             CT_GetFile "gcc-${CT_CC_GCC_VERSION}"                                                               \
                        "http://launchpad.net/gcc-linaro/${linaro_series}/${linaro_version}/+download"       \
-                       https://releases.linaro.org/${YYMM}/components/toolchain/gcc-linaro/${linaro_series} \
+                       https://releases.linaro.org/components/toolchain/gcc-linaro/${linaro_version} \
                        http://cbuild.validation.linaro.org/snapshots
         fi

