diff --git a/ams.lv2/env.ttl b/ams.lv2/env.ttl
deleted file mode 100644
index 357f5ee..0000000
--- a/ams.lv2/env.ttl
+++ /dev/null
@@ -1,111 +0,0 @@
-@prefix doap: 	<http://usefulinc.com/ns/doap#> .
-@prefix foaf: 	<http://xmlns.com/foaf/0.1/> .
-@prefix lv2: 	<http://lv2plug.in/ns/lv2core#> .
-@prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
-@prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
-@prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/env/gui>
-  	a guiext:GtkUI ;
-  	guiext:binary <env_gui.so> .
-
-<http://github.com/blablack/ams-lv2/env>
-	a lv2:Plugin, lv2:EnvelopePlugin ;
-	lv2:project <http://github.com/blablack/ams-lv2/> ;
-	doap:name "AMS Envelope" ;
-	doap:license <http://opensource.org/licenses/isc> ;
-	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/env/gui> ;
-	lv2:optionalFeature lv2:hardRTCapable ;
-	lv2:port [
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 0 ; 
-		lv2:symbol "gate" ;
-		lv2:name "Gate" ;
-		lv2:default 0.0 ;
-		lv2:minimum 0.0 ;
-        lv2:maximum 1.0 ;
-        lv2:portProperty lv2:toggled ;
-	] ,	[
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 1 ; 
-		lv2:symbol "retrigger" ;
-		lv2:name "Retrigger" ;
-		lv2:default 0.0 ;
-		lv2:minimum 0.0 ;
-        lv2:maximum 1.0 ;
-        lv2:portProperty lv2:toggled ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 2 ; 
-		lv2:symbol "attack" ;
-		lv2:name "Attack" ;
-		lv2:default 0.05 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 3 ; 
-		lv2:symbol "decay" ;
-		lv2:name "Decay" ;
-		lv2:default 0.1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] , [
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 4 ; 
-		lv2:symbol "sustain" ;
-		lv2:name "Sustain" ;
-		lv2:default 0.7 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 5 ; 
-		lv2:symbol "release" ;
-		lv2:name "Release" ;
-		lv2:default 0.05 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 6 ; 
-		lv2:symbol "delay" ;
-		lv2:name "Delay" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 7 ; 
-		lv2:symbol "hold" ;
-		lv2:name "Hold" ;
-		lv2:default 0.02 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 8 ; 
-		lv2:symbol "timeScale" ;
-		lv2:name "Time Scale" ;
-		lv2:default 0.1 ;
-		lv2:minimum 1 ;
-		lv2:maximum 10 ;
-	] ,	[
-		a lv2:OutputPort, lv2:CVPort ;
-		lv2:index 9 ; 
-		lv2:symbol "out" ;
-		lv2:name "Out" ;
-		lv2:default 0.0 ;
-		lv2:minimum 0.0 ;
-        lv2:maximum 1.0 ;
-	] ,	[
-		a lv2:OutputPort, lv2:CVPort ;
-		lv2:index 10 ; 
-		lv2:symbol "invOut" ;
-		lv2:name "Inverse Out" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-        lv2:maximum 0.0 ;
-	].
diff --git a/ams.lv2/lfo2_freq.ttl b/ams.lv2/lfo2_freq.ttl
deleted file mode 100644
index cc6611f..0000000
--- a/ams.lv2/lfo2_freq.ttl
+++ /dev/null
@@ -1,82 +0,0 @@
-@prefix doap: 	<http://usefulinc.com/ns/doap#> .
-@prefix foaf: 	<http://xmlns.com/foaf/0.1/> .
-@prefix lv2: 	<http://lv2plug.in/ns/lv2core#> .
-@prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
-@prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
-@prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/lfo2_freq/gui>
-  a guiext:GtkUI ;
-  guiext:binary <lfo2_freq_gui.so> .
-
-<http://github.com/blablack/ams-lv2/lfo2_freq>
-	a lv2:Plugin, lv2:OscillatorPlugin ;
-	lv2:project <http://github.com/blablack/ams-lv2/> ;
-	doap:name "AMS LFO2 - Freq" ;
-	doap:license <http://opensource.org/licenses/isc> ;
-	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/lfo2_freq/gui> ;
-	lv2:optionalFeature lv2:hardRTCapable ;
-	lv2:port [
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 0 ; 
-		lv2:symbol "reset" ;
-		lv2:name "Reset" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-        lv2:maximum 1.0 ;
-	] , [
-		a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 1 ; 
-		lv2:symbol "waveForm" ;
-		lv2:name "Wave Form" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 5 ;
-		lv2:portProperty lv2:enumeration ;
-		lv2:scalePoint
-				[
-			    rdfs:label "Sine" ;
-			    rdf:value 0
-		] ,		[
-				rdfs:label "Triangle" ;
-				rdf:value 1
-		] ,		[
-		  		rdfs:label "Sawtooth Up" ;
-				rdf:value 2
-		] , 	[
-		  		rdfs:label "Sawtooth Down" ;
-				rdf:value 3
-		] ,		[
-		  		rdfs:label "Rectangle" ;
-				rdf:value 4
-		] ,		[
-		  		rdfs:label "S & H" ;
-				rdf:value 5
-		] ;
-	] ,	[
-		a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 2 ; 
-		lv2:symbol "freq" ;
-		lv2:name "Freq" ;
-		lv2:default 5.0 ;
-		lv2:minimum 0.0001 ;
-		lv2:maximum 100.0 ;
-	] ,	[
-		a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 3 ; 
-		lv2:symbol "phi0" ;
-		lv2:name "Phi0" ;
-		lv2:default 0 ;
-		lv2:minimum 0.0 ;
-		lv2:maximum 6.28 ;
-	] ,	[
-		a lv2:OutputPort, lv2:CVPort ;
-		lv2:index 4 ; 
-		lv2:symbol "output" ;
-		lv2:name "Output" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-        lv2:maximum 1.0 ;
-	] .
diff --git a/ams.lv2/vcaexp.ttl b/ams.lv2/vcaexp.ttl
deleted file mode 100644
index ed74338..0000000
--- a/ams.lv2/vcaexp.ttl
+++ /dev/null
@@ -1,92 +0,0 @@
-@prefix doap: 	<http://usefulinc.com/ns/doap#> .
-@prefix foaf: 	<http://xmlns.com/foaf/0.1/> .
-@prefix lv2: 	<http://lv2plug.in/ns/lv2core#> .
-@prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
-@prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
-@prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/vcaexp/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vcaexp_gui.so> .
-
-<http://github.com/blablack/ams-lv2/vcaexp>
-	a lv2:Plugin, lv2:AmplifierPlugin ;
-	lv2:project <http://github.com/blablack/ams-lv2/> ;
-	doap:name "AMS VCA Exp" ;
-	doap:license <http://opensource.org/licenses/isc> ;
-	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vcaexp/gui> ;
-	lv2:optionalFeature lv2:hardRTCapable ;
-	lv2:port [
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 0 ; 
-		lv2:symbol "gain1Data" ;
-		lv2:name "Main Gain" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-        lv2:maximum 1.0 ;
-	] ,	[
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 1 ; 
-		lv2:symbol "gain2Data" ;
-		lv2:name "2nd Gain" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-        lv2:maximum 1.0 ;
-	] ,	[
-		a lv2:InputPort, lv2:AudioPort ;
-		lv2:index 2 ; 
-		lv2:symbol "in1Data" ;
-		lv2:name "In 1" ;
-	] ,	[
-	  	a lv2:InputPort, lv2:AudioPort;
-		lv2:index 3 ; 
-		lv2:symbol "in2Data" ;
-		lv2:name "In 2" ;
-	] , [
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 4 ; 
-		lv2:symbol "gain1" ;
-		lv2:name "Gain Offset" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 5 ; 
-		lv2:symbol "gain2" ;
-		lv2:name "2nd Gain Boost" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] , [
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 6 ; 
-		lv2:symbol "in1" ;
-		lv2:name "In 1 Level" ;
-		lv2:default 1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 2 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 7 ; 
-		lv2:symbol "in2" ;
-		lv2:name "In 2 Level" ;
-		lv2:default 1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 2 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 8 ; 
-		lv2:symbol "outputLevel" ;
-		lv2:name "Output Level" ;
-		lv2:default 1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 2 ;
-	] ,	[
-		a lv2:OutputPort, lv2:AudioPort ;
-		lv2:index 9 ; 
-		lv2:symbol "out" ;
-		lv2:name "Out" ;
-	] .
diff --git a/ams.lv2/vcalin.ttl b/ams.lv2/vcalin.ttl
deleted file mode 100644
index 24e9d44..0000000
--- a/ams.lv2/vcalin.ttl
+++ /dev/null
@@ -1,92 +0,0 @@
-@prefix doap: 	<http://usefulinc.com/ns/doap#> .
-@prefix foaf: 	<http://xmlns.com/foaf/0.1/> .
-@prefix lv2: 	<http://lv2plug.in/ns/lv2core#> .
-@prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
-@prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
-@prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/vcalin/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vcalin_gui.so> .
-
-<http://github.com/blablack/ams-lv2/vcalin>
-	a lv2:Plugin, lv2:AmplifierPlugin ;
-	lv2:project <http://github.com/blablack/ams-lv2/> ;
-	doap:name "AMS VCA Lin" ;
-	doap:license <http://opensource.org/licenses/isc> ;
-	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vcalin/gui> ;
-	lv2:optionalFeature lv2:hardRTCapable ;
-	lv2:port [
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 0 ; 
-		lv2:symbol "gain1Data" ;
-		lv2:name "Main Gain" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-        lv2:maximum 1.0 ;
-	] ,	[
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 1 ; 
-		lv2:symbol "gain2Data" ;
-		lv2:name "2nd Gain" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-        lv2:maximum 1.0 ;
-	] ,	[
-		a lv2:InputPort, lv2:AudioPort ;
-		lv2:index 2 ; 
-		lv2:symbol "in1Data" ;
-		lv2:name "In 1 Mod" ;
-	] ,	[
-	  	a lv2:InputPort, lv2:AudioPort ;
-		lv2:index 3 ; 
-		lv2:symbol "in2Data" ;
-		lv2:name "In 2 Mod" ;
-	] , [
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 4 ; 
-		lv2:symbol "gain1" ;
-		lv2:name "Gain Offset" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 5 ; 
-		lv2:symbol "gain2" ;
-		lv2:name "2nd Gain Boost" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] , [
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 6 ; 
-		lv2:symbol "in1" ;
-		lv2:name "In 1" ;
-		lv2:default 1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 2 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 7 ; 
-		lv2:symbol "in2" ;
-		lv2:name "In 2" ;
-		lv2:default 1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 2 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 8 ; 
-		lv2:symbol "outputLevel" ;
-		lv2:name "Output Level" ;
-		lv2:default 1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 2 ;
-	] ,	[
-		a lv2:OutputPort, lv2:AudioPort ;
-		lv2:index 9 ; 
-		lv2:symbol "out" ;
-		lv2:name "Out" ;
-	] .
diff --git a/ams.lv2/vcf.ttl b/ams.lv2/vcf.ttl
deleted file mode 100644
index 4aea079..0000000
--- a/ams.lv2/vcf.ttl
+++ /dev/null
@@ -1,147 +0,0 @@
-@prefix doap: 	<http://usefulinc.com/ns/doap#> .
-@prefix foaf: 	<http://xmlns.com/foaf/0.1/> .
-@prefix lv2: 	<http://lv2plug.in/ns/lv2core#> .
-@prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
-@prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
-@prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/vcf/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vcf_gui.so> .
-
-<http://github.com/blablack/ams-lv2/vcf>
-	a lv2:Plugin, lv2:SpectralPlugin ;
-	lv2:project <http://github.com/blablack/ams-lv2/> ;
-	doap:name "AMS VCF" ;
-	doap:license <http://opensource.org/licenses/isc> ;
-	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vcf/gui> ;
-	lv2:optionalFeature lv2:hardRTCapable ;
-	lv2:port 
-		[
-		a lv2:InputPort, lv2:AudioPort;
-		lv2:index 0 ; 
-		lv2:symbol "in" ;
-		lv2:name "In" ;
-	] ,	[
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 1 ; 
-		lv2:symbol "freqPort" ;
-		lv2:name "Frequency Mod" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-	        lv2:maximum 1.0 ;
-	] ,	[
-		a lv2:InputPort, lv2:CVPort ;
-		lv2:index 2 ; 
-		lv2:symbol "expFM" ;
-		lv2:name "Exp FM" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-	        lv2:maximum 1.0 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:CVPort ;
-		lv2:index 3 ; 
-		lv2:symbol "linFM" ;
-		lv2:name "Lin FM" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-	        lv2:maximum 1.0 ;
-	] , 	[
-	  	a lv2:InputPort, lv2:CVPort ;
-		lv2:index 4 ; 
-		lv2:symbol "resonancePort" ;
-		lv2:name "Resonance Mod" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-	        lv2:maximum 1.0 ;
-	] , 	[
-		a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 5 ; 
-		lv2:symbol "vcfType" ;
-		lv2:name "VCF Type" ;
-		lv2:default 0;
-		lv2:minimum 0;
-		lv2:maximum 7;
-		lv2:portProperty lv2:enumeration ;
-		lv2:scalePoint
-				[
-			    	rdfs:label "Resonant Lowpass" ;
-			   	rdf:value 0 ;
-			] ,	[
-				rdfs:label "Lowpass" ;
-				rdf:value 1 ;
-			] ,	[
-		  		rdfs:label "Highpass" ;
-				rdf:value 2 ;
-			] , 	[
-		  		rdfs:label "Bandpass I" ;
-				rdf:value 3 ;
-			] ,	[
-		  		rdfs:label "Bandpass II" ;
-				rdf:value 4 ;
-			] ,	[
-		  		rdfs:label "Notch" ;
-				rdf:value 5 ;
-			] , 	[
-		  		rdfs:label "24 dB Lowpass I" ;
-				rdf:value 6 ;
-			] , 	[
-		  		rdfs:label "24 dB Lowpass II" ;
-				rdf:value 7 ;
-			]
-	] , 	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 6 ; 
-		lv2:symbol "inputGain" ;
-		lv2:name "Input Gain" ;
-		lv2:default 1 ;
-		lv2:minimum 0 ;
-		lv2:maximum 10 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 7 ; 
-		lv2:symbol "freq" ;
-		lv2:name "Frequency" ;
-		lv2:default 5 ;
-		lv2:minimum 0 ;
-		lv2:maximum 10 ;
-	] , 	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 8 ; 
-		lv2:symbol "expFMGain" ;
-		lv2:name "Exp. FM Gain" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 10 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 9 ; 
-		lv2:symbol "linFMGain" ;
-		lv2:name "Lin. FM Gain" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 10 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 10 ; 
-		lv2:symbol "resonance" ;
-		lv2:name "Resonance" ;
-		lv2:default 0.8 ;
-		lv2:minimum 0.01 ;
-		lv2:maximum 1.0 ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 11 ; 
-		lv2:symbol "resonanceGain" ;
-		lv2:name "Resonance Gain" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-	] ,	[
-		a lv2:OutputPort, lv2:AudioPort ;
-		lv2:index 12 ; 
-		lv2:symbol "out" ;
-		lv2:name "Out" ;
-	] .
diff --git a/ams.lv2/vco3.ttl b/ams.lv2/vco3.ttl
deleted file mode 100644
index daf545c..0000000
--- a/ams.lv2/vco3.ttl
+++ /dev/null
@@ -1,207 +0,0 @@
-@prefix doap: 	<http://usefulinc.com/ns/doap#> .
-@prefix foaf: 	<http://xmlns.com/foaf/0.1/> .
-@prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
-@prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
-@prefix lv2: 	<http://lv2plug.in/ns/lv2core#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-@prefix pg:     <http://lv2plug.in/ns/ext/port-groups#> .
-@prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix param:  <http://lv2plug.in/ns/ext/parameters#> .
-
-<http://github.com/blablack/ams-lv2/vco3/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vco3_gui.so> .
-
-<http://github.com/blablack/ams-lv2/vco3#group_frequency>
-    a param:OscillatorControls ;
-    lv2:name "Frequency" ;
-    lv2:symbol "grpFrequency" .
-
-<http://github.com/blablack/ams-lv2/vco3#group_pw_phase>
-    a param:OscillatorControls ;
-    lv2:name "Pulse Width - Phase" ;
-    lv2:symbol "grpPWPhase" .
-
-<http://github.com/blablack/ams-lv2/vco3#group_modulation>
-    a param:OscillatorControls ;
-    lv2:name "Modulation" ;
-    lv2:symbol "grpModulation" .
-
-<http://github.com/blablack/ams-lv2/vco3#group_input>
-    a param:OscillatorControls, pg:InputGroup ;
-    lv2:name "Input" ;
-    lv2:symbol "grpInput" .
-
-<http://github.com/blablack/ams-lv2/vco3>
-	a lv2:Plugin ,
-		lv2:OscillatorPlugin ;
-	lv2:project <http://github.com/blablack/ams-lv2/> ;
-	doap:name "AMS VCO3" ;
-	doap:license <http://opensource.org/licenses/isc> ;
-	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vco3/gui> ;
-	lv2:optionalFeature lv2:hardRTCapable ;
-	lv2:port [
-		a lv2:CVPort, lv2:InputPort ;
-		lv2:index 0 ; 
-		lv2:symbol "freq" ;
-		lv2:name "Frequency";
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-		lv2:maximum 1.0 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupInput> ;
-	] ,	[
-		a lv2:CVPort, lv2:InputPort ;
-		lv2:index 1 ; 
-		lv2:symbol "expFM" ;
-		lv2:name "Exp FM" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-		lv2:maximum 1.0 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupInput> ;
-	] ,	[
-		a lv2:CVPort, lv2:InputPort ;
-		lv2:index 2 ; 
-		lv2:symbol "linFM" ;
-		lv2:name "Lin FM" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-		lv2:maximum 1.0 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupInput> ;
-	] ,	[
-	  	a lv2:CVPort, lv2:InputPort ;
-		lv2:index 3 ; 
-		lv2:symbol "pwPort" ;
-		lv2:name "PW" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-		lv2:maximum 1.0 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupInput> ;
-	] ,	[
-	  	a lv2:CVPort, lv2:InputPort ;
-		lv2:index 4 ; 
-		lv2:symbol "sync" ;
-		lv2:name "Sync" ;
-		lv2:default 0.0 ;
-		lv2:minimum -1.0 ;
-		lv2:maximum 1.0 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupInput> ;
-	] ,	[
-		a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 5 ; 
-		lv2:symbol "waveForm" ;
-		lv2:name "Wave Form" ;
-		lv2:default 0;
-		lv2:minimum 0;
-		lv2:maximum 6;
-		lv2:portProperty lv2:enumeration ;
-		lv2:scalePoint
-				[
-			    rdfs:label "Sine" ;
-			    rdf:value 0
-		] ,		[
-				rdfs:label "Triangle" ;
-				rdf:value 1
-		] ,		[
-		  		rdfs:label "Sawtooth" ;
-				rdf:value 2
-		] , 	[
-		  		rdfs:label "Rectangle" ;
-				rdf:value 3
-		] ,		[
-		  		rdfs:label "Aux Saw 1" ;
-				rdf:value 4
-		] ,		[
-		  		rdfs:label "Aux Saw 2" ;
-				rdf:value 5
-		] , 	[
-		  		rdfs:label "Aux Saw 3" ;
-				rdf:value 6
-		]
-	] , [
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 6 ; 
-		lv2:symbol "octave" ;
-		lv2:name "Octave" ;
-		lv2:default 3 ;
-		lv2:minimum 0 ;
-		lv2:maximum 6 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupFrequency> ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 7 ; 
-		lv2:symbol "tune" ;
-		lv2:name "Tune" ;
-		lv2:default 0 ;
-		lv2:minimum -1 ;
-		lv2:maximum 1 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupFrequency> ;
-	] , [
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 8 ; 
-		lv2:symbol "semitone" ;
-		lv2:name "Semitone" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 12 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupFrequency> ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 9 ; 
-		lv2:symbol "pw0" ;
-		lv2:name "PW0" ;
-		lv2:default 0.5 ;
-		lv2:minimum 0.1 ;
-		lv2:maximum 0.9 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupPWPhase> ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 10 ; 
-		lv2:symbol "pwGain" ;
-		lv2:name "PW Gain" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 1 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupPWPhase> ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 11 ; 
-		lv2:symbol "edge" ;
-		lv2:name "Edge" ;
-		lv2:default 1 ;
-		lv2:minimum 1 ;
-		lv2:maximum 10 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupPWPhase> ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 12 ; 
-		lv2:symbol "phi0" ;
-		lv2:name "Phi0" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 6.283 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupPWPhase> ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 13 ; 
-		lv2:symbol "expFMGain" ;
-		lv2:name "Exp FM Gain" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 10 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupModulation> ;
-	] ,	[
-	  	a lv2:InputPort, lv2:ControlPort ;
-		lv2:index 14 ; 
-		lv2:symbol "linFMGain" ;
-		lv2:name "Lin FM Gain" ;
-		lv2:default 0 ;
-		lv2:minimum 0 ;
-		lv2:maximum 10 ;
-		pg:group <http://github.com/blablack/ams-lv2/vco3#groupModulation> ;
-	] ,	[
-		a lv2:AudioPort , lv2:OutputPort ;
-		lv2:index 15 ; 
-		lv2:symbol "out" ;
-		lv2:name "Out"
-	] .
diff --git a/ams.lv2/manifest.ttl b/ams.lv2/manifest.ttl
index 38cd688..3c0f44f 100644
--- a/ams.lv2/manifest.ttl
+++ b/ams.lv2/manifest.ttl
@@ -111,11 +111,6 @@
 	lv2:binary <dynamicwaves2_8.so>  ;
 	rdfs:seeAlso <dynamicwaves2_8.ttl> .
 
-<http://github.com/blablack/ams-lv2/env>
-	a lv2:Plugin ;
-	lv2:binary <env.so>  ;
-	rdfs:seeAlso <env.ttl> .
-
 <http://github.com/blablack/ams-lv2/fftvocoder>
 	a lv2:Plugin ;
 	lv2:binary <fftvocoder.so>  ;
@@ -146,11 +141,6 @@
 	lv2:binary <lfo2_tempo.so>  ;
 	rdfs:seeAlso <lfo2_tempo.ttl> .
 
-<http://github.com/blablack/ams-lv2/lfo2_freq>
-	a lv2:Plugin ;
-	lv2:binary <lfo2_freq.so>  ;
-	rdfs:seeAlso <lfo2_freq.ttl> .
-
 <http://github.com/blablack/ams-lv2/lfo>
 	a lv2:Plugin ;
 	lv2:binary <lfo.so>  ;
@@ -256,16 +246,6 @@
 	lv2:binary <transport.so>  ;
 	rdfs:seeAlso <transport.ttl> .
 
-<http://github.com/blablack/ams-lv2/vcaexp>
-	a lv2:Plugin ;
-	lv2:binary <vcaexp.so>  ;
-	rdfs:seeAlso <vcaexp.ttl> .
-
-<http://github.com/blablack/ams-lv2/vcalin>
-	a lv2:Plugin ;
-	lv2:binary <vcalin.so>  ;
-	rdfs:seeAlso <vcalin.ttl> .
-
 <http://github.com/blablack/ams-lv2/vcdelay>
 	a lv2:Plugin ;
 	lv2:binary <vcdelay.so>  ;
@@ -286,21 +266,11 @@
 	lv2:binary <vcenvii.so>  ;
 	rdfs:seeAlso <vcenvii.ttl> .
 
-<http://github.com/blablack/ams-lv2/vcf>
-	a lv2:Plugin ;
-	lv2:binary <vcf.so>  ;
-	rdfs:seeAlso <vcf.ttl> .
-
 <http://github.com/blablack/ams-lv2/vco2>
 	a lv2:Plugin ;
 	lv2:binary <vco2.so>  ;
 	rdfs:seeAlso <vco2.ttl> .
 
-<http://github.com/blablack/ams-lv2/vco3>
-	a lv2:Plugin ;
-	lv2:binary <vco3.so>  ;
-	rdfs:seeAlso <vco3.ttl> .
-
 <http://github.com/blablack/ams-lv2/vcorgan_4>
 	a lv2:Plugin ;
 	lv2:binary <vcorgan_4.so>  ;
diff --git a/wscript b/wscript
index d61e72a..b0589e0 100644
--- a/wscript
+++ b/wscript
@@ -132,13 +132,11 @@ def build(bld):
     cvs
     cvtocontrol
     delay
-    env
     haas
     hysteresis
     hztovc
     inv
     lfo2_tempo
-    lfo2_freq
     lfo
     mixer_8ch
     mixer_4ch
@@ -154,7 +152,6 @@ def build(bld):
     stereomixer_4ch
     stereomixer_8ch
     transport
-    vcalin
     vcdelay
     vcenv
     vcpanning
@@ -188,12 +185,9 @@ def build(bld):
 ########################################################################
 
     plugins = '''
-    vcaexp
     vcdoubledecay
     vcenvii
-    vcf
     vco2
-    vco3
     '''.split()
 
     for i in plugins:
