diff --git a/modules/JUCE/extras/Build/CMake/JUCEUtils.cmake b/modules/JUCE/extras/Build/CMake/JUCEUtils.cmake
index 0d092b2e0..3b4bf5105 100644
--- a/modules/JUCE/extras/Build/CMake/JUCEUtils.cmake
+++ b/modules/JUCE/extras/Build/CMake/JUCEUtils.cmake
@@ -1367,34 +1367,6 @@ function(_juce_set_plugin_target_properties shared_code_target kind)
                 LIBRARY_OUTPUT_DIRECTORY "${output_path}")
         endif()
 
-
-       # LV2 tools dont work with a space in the .so file
-       set(lv2_shared_lib_property "$<TARGET_PROPERTY:${shared_code_target},JUCE_LV2_SHARED_LIBRARY_NAME>")
-       set(lv2_shared_lib $<IF:$<BOOL:${lv2_shared_lib_property}>,${lv2_shared_lib_property},${product_name}> )
-        add_custom_command(TARGET ${target_name} POST_BUILD
-         COMMAND cmake -E rename ${product_name}.so ${lv2_shared_lib}.so
-          WORKING_DIRECTORY "${products_folder}/${product_name}.lv2/"
-          VERBATIM)
-       
-       # generate .ttl files
-        if(LV2_TTL_GENERATOR)
-            add_custom_command(TARGET ${target_name} POST_BUILD
-                COMMAND ${LV2_TTL_GENERATOR} "./${lv2_shared_lib}.so"
-                WORKING_DIRECTORY "${products_folder}/${product_name}.lv2/"
-                VERBATIM)
-        else()
-            add_executable(${target_name}_lv2_ttl_generator ${JUCE_SOURCE_DIR}/extras/Build/lv2_ttl_generator/lv2_ttl_generator.c)
-            set_source_files_properties(${JUCE_SOURCE_DIR}/extras/Build/lv2_ttl_generator/lv2_ttl_generator.c PROPERTIES LANGUAGE CXX)
-            target_link_libraries(${target_name}_lv2_ttl_generator dl)
-            add_custom_command(TARGET ${target_name} POST_BUILD
-                COMMAND ${target_name}_lv2_ttl_generator "./${lv2_shared_lib}.so"
-                DEPENDS ${target_name} lv2_ttl_generator
-                WORKING_DIRECTORY "${products_folder}/${product_name}.lv2/"
-                VERBATIM)
-        endif()
-
-        _juce_copy_after_build(${shared_code_target} ${target_name} "${output_path}" JUCE_LV2_COPY_DIR)
-
     elseif(kind STREQUAL "VST")
         set_target_properties(${target_name} PROPERTIES
             BUNDLE_EXTENSION vst
