diff --git a/toolchain/helpers.mk b/toolchain/helpers.mk
index ee878e8..581996a 100644
--- a/toolchain/helpers.mk
+++ b/toolchain/helpers.mk
@@ -354,7 +354,7 @@ check_uclibc = \
 check_arm_abi = \
 	__CROSS_CC=$(strip $1) ; \
 	__CROSS_READELF=$(strip $2) ; \
-	EXT_TOOLCHAIN_TARGET=`LANG=C $${__CROSS_CC} -v 2>&1 | grep ^Target | cut -f2 -d ' '` ; \
+	EXT_TOOLCHAIN_TARGET=`LANG=C $${__CROSS_CC} -v 2>&1 | grep ^Target | cut -f2 -d ' ' | awk 'sub(".static","")'` ; \
 	if ! echo $${EXT_TOOLCHAIN_TARGET} | grep -qE 'eabi(hf)?$$' ; then \
 		echo "External toolchain uses the unsuported OABI" ; \
 		exit 1 ; \
