diff --git a/library/cmake/SfizzDeps.cmake b/library/cmake/SfizzDeps.cmake
index fbf5f8e0..5f95eafb 100644
--- a/library/cmake/SfizzDeps.cmake
+++ b/library/cmake/SfizzDeps.cmake
@@ -231,9 +231,6 @@ if(JACK_FOUND)
     link_directories(${JACK_LIBRARY_DIRS})
 endif()
 
-# The Qt library (CaptureEG devtool)
-find_package(Qt5 COMPONENTS Widgets)
-
 # The fmidi library
 add_library(sfizz_fmidi STATIC
     "external/fmidi/sources/fmidi/fmidi.h"
