diff --git a/kernel/cycle.h b/kernel/cycle.h
index fe3dd50..95fbefd 100644
--- a/kernel/cycle.h
+++ b/kernel/cycle.h
@@ -515,6 +515,18 @@ INLINE_ELAPSED(inline)
 #endif
 #endif /* HAVE_MIPS_ZBUS_TIMER */
 
+#if defined(__ARM_ARCH_7A__)
+typedef uint64_t ticks;
+static inline ticks getticks(void)
+{
+  uint32_t r;
+  asm volatile("mrc p15, 0, %0, c9, c13, 0\t\n" : "=r"(r)); /* Read PMCCNTR       */
+  return ((uint64_t)r) << 6;                                /* 1 tick = 64 clocks */
+}
+INLINE_ELAPSED(inline)
+#define HAVE_TICK_COUNTER
+#endif
+
 #if defined(HAVE_ARMV7A_CNTVCT)
 typedef uint64_t ticks;
 static inline ticks getticks(void)
