diff --git a/src/fsck/fsck.c b/src/fsck/fsck.c
index 78ceeb6..7c262ae 100644
--- a/src/fsck/fsck.c
+++ b/src/fsck/fsck.c
@@ -322,13 +322,6 @@ int main(int argc, char *argv[]) {
         cmdline[i++] =  arg_repair;
         cmdline[i++] = "-T";
 
-        /*
-         * Since util-linux v2.25 fsck uses /run/fsck/<diskname>.lock files.
-         * The previous versions use flock for the device and conflict with
-         * udevd, see https://bugs.freedesktop.org/show_bug.cgi?id=79576#c5
-         */
-        cmdline[i++] = "-l";
-
         if (!root_directory)
                 cmdline[i++] = "-M";
 
