diff --git a/src/core/unit.c b/src/core/unit.c
index ee8e607..34f57a2 100644
--- a/src/core/unit.c
+++ b/src/core/unit.c
@@ -1653,10 +1653,7 @@ static void unit_check_binds_to(Unit *u) {
         if (!stop)
                 return;
 
-        log_unit_info(u->id, "Unit %s is bound to inactive unit. Stopping, too.", u->id);
-
-        /* A unit we need to run is gone. Sniff. Let's stop this. */
-        manager_add_job(u->manager, JOB_STOP, u, JOB_FAIL, true, NULL, NULL);
+        log_unit_warning(u->id, "Unit %s is bound to inactive unit. Ignored.", u->id);
 }
 
 static void retroactively_start_dependencies(Unit *u) {
