diff --git a/Makefile b/Makefile
index fdf0cae..714f5d7 100644
--- a/Makefile
+++ b/Makefile
@@ -14,8 +14,8 @@ RW=$(abspath $(ROBTK))/
 export RW
 endif
 
-ifneq ($(MOD),)
-  SUBDIRS = b_whirl b_synth
+ifneq ($(MOD_BUILD),)
+  SUBDIRS = b_whirl b_synth b_overdrive b_reverb
 else
   SUBDIRS = b_overdrive b_whirl b_reverb b_conv b_chorato src b_synth ui
 endif
diff --git a/common.mak b/common.mak
index fb8ad0d..215a3c0 100644
--- a/common.mak
+++ b/common.mak
@@ -106,7 +106,7 @@ ifeq ($(ENABLE_CONVOLUTION), yes)
   CC=$(CXX)
 endif
 
-ifneq ($(MOD),)
+ifneq ($(MOD_BUILD),)
 	MODBRAND=mod:brand \"x42\";
 	MODLABEL=mod:label \"$(MODNAME)\";
 	MODGUITTL=, <modgui.ttl>
@@ -128,7 +128,7 @@ else
   endif
 endif
 
-ifeq ($(MOD),)
+ifeq ($(MOD_BUILD),)
   HAVE_UI=$(shell pkg-config --exists $(PKG_GL_LIBS) ftgl && echo $(FONT_FOUND))
 else
   HAVE_UI=no
