diff --git a/plugins/LV2/CMakeLists.txt b/plugins/LV2/CMakeLists.txt
index 281de56..5db4e5e 100755
--- a/plugins/LV2/CMakeLists.txt
+++ b/plugins/LV2/CMakeLists.txt
@@ -7,6 +7,10 @@ add_compile_options(-Wno-unused-but-set-variable)   # AQuickVoiceClip
 add_compile_options(-Wno-misleading-indentation)    # curve
 add_compile_options(-Wno-strict-aliasing)           # DustBunny
 
+# FIXME read from env var
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ffast-math -fsingle-precision-constant")
+set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-O1 -Wl,--as-needed -Wl,--strip-all")
+
 configure_file(manifest.ttl.in airwindows.lv2/manifest.ttl @ONLY)
 
 include(Helpers.cmake)
