diff --git a/plugins/LV2/CMakeLists.txt b/plugins/LV2/CMakeLists.txt
index 281de56..8cce1ab 100755
--- a/plugins/LV2/CMakeLists.txt
+++ b/plugins/LV2/CMakeLists.txt
@@ -14,87 +14,87 @@ include(Helpers.cmake)
 include_directories(include)
 
 # add_subdirectory(include/vstsdk)
-add_airwindows_plugin(Acceleration)
-add_airwindows_plugin(ADClip7)
-add_airwindows_plugin(ADT)
-add_airwindows_plugin(Air)
-add_airwindows_plugin(Apicolypse)
-add_airwindows_plugin(AQuickVoiceClip)
-add_airwindows_plugin(AtmosphereBuss)
-add_airwindows_plugin(AtmosphereChannel)
-add_airwindows_plugin(Aura)
-add_airwindows_plugin(Average)
-add_airwindows_plugin(AverMatrix)
+#add_airwindows_plugin(Acceleration)
+#add_airwindows_plugin(ADClip7)
+#add_airwindows_plugin(ADT)
+#add_airwindows_plugin(Air)
+#add_airwindows_plugin(Apicolypse)
+#add_airwindows_plugin(AQuickVoiceClip)
+#add_airwindows_plugin(AtmosphereBuss)
+#add_airwindows_plugin(AtmosphereChannel)
+#add_airwindows_plugin(Aura)
+#add_airwindows_plugin(Average)
+#add_airwindows_plugin(AverMatrix)
 add_airwindows_plugin(BassDrive)
-add_airwindows_plugin(BassKit)
-add_airwindows_plugin(Baxandall)
-add_airwindows_plugin(Beam)
-add_airwindows_plugin(Biquad)
-add_airwindows_plugin(Biquad2)
-add_airwindows_plugin(BiquadOneHalf)
-add_airwindows_plugin(Bite)
-add_airwindows_plugin(BitGlitter)
-add_airwindows_plugin(BitShiftGain)
-add_airwindows_plugin(BlockParty)
-add_airwindows_plugin(BrassRider)
-add_airwindows_plugin(BrightAmbience)
-add_airwindows_plugin(BuildATPDF)
-add_airwindows_plugin(BussColors4)
-add_airwindows_plugin(ButterComp)
-add_airwindows_plugin(ButterComp2)
-add_airwindows_plugin(C5RawBuss)
-add_airwindows_plugin(C5RawChannel)
-add_airwindows_plugin(Calibre)
-add_airwindows_plugin(Capacitor)
-add_airwindows_plugin(Channel4)
-add_airwindows_plugin(Channel5)
-add_airwindows_plugin(Channel6)
-add_airwindows_plugin(Channel7)
-add_airwindows_plugin(Chorus)
-add_airwindows_plugin(ChorusEnsemble)
-add_airwindows_plugin(ChromeOxide)
-add_airwindows_plugin(Cider)
-add_airwindows_plugin(ClipOnly)
-add_airwindows_plugin(Coils)
-add_airwindows_plugin(Cojones)
-add_airwindows_plugin(Compresaturator)
+#add_airwindows_plugin(BassKit)
+#add_airwindows_plugin(Baxandall)
+#add_airwindows_plugin(Beam)
+#add_airwindows_plugin(Biquad)
+#add_airwindows_plugin(Biquad2)
+#add_airwindows_plugin(BiquadOneHalf)
+#add_airwindows_plugin(Bite)
+#add_airwindows_plugin(BitGlitter)
+#add_airwindows_plugin(BitShiftGain)
+#add_airwindows_plugin(BlockParty)
+#add_airwindows_plugin(BrassRider)
+#add_airwindows_plugin(BrightAmbience)
+#add_airwindows_plugin(BuildATPDF)
+#add_airwindows_plugin(BussColors4)
+#add_airwindows_plugin(ButterComp)
+#add_airwindows_plugin(ButterComp2)
+#add_airwindows_plugin(C5RawBuss)
+#add_airwindows_plugin(C5RawChannel)
+#add_airwindows_plugin(Calibre)
+#add_airwindows_plugin(Capacitor)
+#add_airwindows_plugin(Channel4)
+#add_airwindows_plugin(Channel5)
+#add_airwindows_plugin(Channel6)
+#add_airwindows_plugin(Channel7)
+#add_airwindows_plugin(Chorus)
+#add_airwindows_plugin(ChorusEnsemble)
+#add_airwindows_plugin(ChromeOxide)
+#add_airwindows_plugin(Cider)
+#add_airwindows_plugin(ClipOnly)
+#add_airwindows_plugin(Coils)
+#add_airwindows_plugin(Cojones)
+#add_airwindows_plugin(Compresaturator)
 # add_airwindows_plugin(Console4Buss)
 # add_airwindows_plugin(Console4Channel)
-add_airwindows_plugin(Console5Buss)
-add_airwindows_plugin(Console5Channel)
-add_airwindows_plugin(Console5DarkCh)
-add_airwindows_plugin(Console6Buss)
-add_airwindows_plugin(Console6Channel)
-add_airwindows_plugin(CrunchyGrooveWear)
-add_airwindows_plugin(Crystal)
-add_airwindows_plugin(CStrip)
-add_airwindows_plugin(curve)
-add_airwindows_plugin(Dark)
-add_airwindows_plugin(DCVoltage)
-add_airwindows_plugin(Deckwrecka)
-add_airwindows_plugin(DeBess)
-add_airwindows_plugin(DeEss)
-add_airwindows_plugin(DeHiss)
-add_airwindows_plugin(Density)
-add_airwindows_plugin(DeRez)
-add_airwindows_plugin(DeRez2)
-add_airwindows_plugin(Desk)
-add_airwindows_plugin(Desk4)
-add_airwindows_plugin(DigitalBlack)
-add_airwindows_plugin(Distance)
-add_airwindows_plugin(Distance2)
-add_airwindows_plugin(Distortion)
-add_airwindows_plugin(DoublePaul)
-add_airwindows_plugin(Ditherbox)
-add_airwindows_plugin(DitherFloat)
-add_airwindows_plugin(DitherMeDiskers)
-add_airwindows_plugin(DitherMeTimbers)
-add_airwindows_plugin(Drive)
-add_airwindows_plugin(DrumSlam)
-add_airwindows_plugin(DubCenter)
-add_airwindows_plugin(DubSub)
-add_airwindows_plugin(DustBunny)
-add_airwindows_plugin(Dyno)
+#add_airwindows_plugin(Console5Buss)
+#add_airwindows_plugin(Console5Channel)
+#add_airwindows_plugin(Console5DarkCh)
+#add_airwindows_plugin(Console6Buss)
+#add_airwindows_plugin(Console6Channel)
+#add_airwindows_plugin(CrunchyGrooveWear)
+#add_airwindows_plugin(Crystal)
+#add_airwindows_plugin(CStrip)
+#add_airwindows_plugin(curve)
+#add_airwindows_plugin(Dark)
+#add_airwindows_plugin(DCVoltage)
+#add_airwindows_plugin(Deckwrecka)
+#add_airwindows_plugin(DeBess)
+#add_airwindows_plugin(DeEss)
+#add_airwindows_plugin(DeHiss)
+#add_airwindows_plugin(Density)
+#add_airwindows_plugin(DeRez)
+#add_airwindows_plugin(DeRez2)
+#add_airwindows_plugin(Desk)
+#add_airwindows_plugin(Desk4)
+#add_airwindows_plugin(DigitalBlack)
+#add_airwindows_plugin(Distance)
+#add_airwindows_plugin(Distance2)
+#add_airwindows_plugin(Distortion)
+#add_airwindows_plugin(DoublePaul)
+#add_airwindows_plugin(Ditherbox)
+#add_airwindows_plugin(DitherFloat)
+#add_airwindows_plugin(DitherMeDiskers)
+#add_airwindows_plugin(DitherMeTimbers)
+#add_airwindows_plugin(Drive)
+#add_airwindows_plugin(DrumSlam)
+#add_airwindows_plugin(DubCenter)
+#add_airwindows_plugin(DubSub)
+#add_airwindows_plugin(DustBunny)
+#add_airwindows_plugin(Dyno)
 # add_airwindows_plugin(EQ)
 # add_airwindows_plugin(EdIsDim)
 # add_airwindows_plugin(Elation)
diff --git a/plugins/LV2/manifest.ttl.in b/plugins/LV2/manifest.ttl.in
index 072d235..a354ab5 100644
--- a/plugins/LV2/manifest.ttl.in
+++ b/plugins/LV2/manifest.ttl.in
@@ -25,555 +25,9 @@ airwindows:
         foaf:mbox <chrisj@airwindows.com>
     ] .
 
-airwindows:acceleration
-    a lv2:Plugin ;
-    doap:name "Airwindows Acceleration Limiter" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Acceleration@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Acceleration.ttl> .
-
-airwindows:adclip-7
-    a lv2:Plugin ;
-    doap:name "Airwindows ADClip7 Loudness Maximizer" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <ADClip7@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <ADClip7.ttl> .
-
-airwindows:adt
-    a lv2:Plugin ;
-    doap:name "Airwindows Adjustable Delay Taps" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <ADT@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <ADT.ttl> .
-
-airwindows:air
-    a lv2:Plugin ;
-    doap:name "Airwindows Air" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Air@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Air.ttl> .
-
-airwindows:apicolypse
-    a lv2:Plugin ;
-    doap:name "Airwindows Apicolypse" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Apicolypse@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Apicolypse.ttl> .
-
-airwindows:aquickvoiceclip
-    a lv2:Plugin ;
-    doap:name "Airwindows AQuickVoiceClip" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <AQuickVoiceClip@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <AQuickVoiceClip.ttl> .
-
-airwindows:atmospherebuss
-    a lv2:Plugin ;
-    doap:name "Airwindows AtmosphereBuss" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <AtmosphereBuss@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <AtmosphereBuss.ttl> .
-
-airwindows:atmospherechannel
-    a lv2:Plugin ;
-    doap:name "Airwindows AtmosphereChannel" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <AtmosphereChannel@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <AtmosphereChannel.ttl> .
-
-airwindows:aura
-    a lv2:Plugin ;
-    doap:name "Airwindows Aura" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Aura@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Aura.ttl> .
-
-airwindows:average
-    a lv2:Plugin ;
-    doap:name "Airwindows Average" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Average@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Average.ttl> .
-
-airwindows:avermatrix
-    a lv2:Plugin ;
-    doap:name "Airwindows AverMatrix" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <AverMatrix@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <AverMatrix.ttl> .
-
 airwindows:bassdrive
     a lv2:Plugin ;
     doap:name "Airwindows BassDrive" ;
     doap:license <http://opensource.org/licenses/mit> ;
     lv2:binary <BassDrive@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
     rdfs:seeAlso <BassDrive.ttl> .
-
-airwindows:basskit
-    a lv2:Plugin ;
-    doap:name "Airwindows BassKit" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BassKit@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BassKit.ttl> .
-
-airwindows:baxandall
-    a lv2:Plugin ;
-    doap:name "Airwindows Baxandall" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Baxandall@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Baxandall.ttl> .
-
-airwindows:beam
-    a lv2:Plugin ;
-    doap:name "Airwindows Beam" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Beam@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Beam.ttl> .
-
-airwindows:biquad
-    a lv2:Plugin ;
-    doap:name "Airwindows Biquad" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Biquad@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Biquad.ttl> .
-
-airwindows:biquad2
-    a lv2:Plugin ;
-    doap:name "Airwindows Biquad2" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Biquad2@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Biquad2.ttl> .
-
-airwindows:biquadonehalf
-    a lv2:Plugin ;
-    doap:name "Airwindows BiquadOneHalf" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BiquadOneHalf@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BiquadOneHalf.ttl> .
-
-airwindows:bitglitter
-    a lv2:Plugin ;
-    doap:name "Airwindows BitGlitter" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BitGlitter@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BitGlitter.ttl> .
-
-airwindows:bitshiftgain
-    a lv2:Plugin ;
-    doap:name "Airwindows BitShiftGain" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BitShiftGain@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BitShiftGain.ttl> .
-
-airwindows:bite
-    a lv2:Plugin ;
-    doap:name "Airwindows Bite" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Bite@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Bite.ttl> .
-
-airwindows:blockparty
-    a lv2:Plugin ;
-    doap:name "Airwindows BlockParty" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BlockParty@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BlockParty.ttl> .
-
-airwindows:brassrider
-    a lv2:Plugin ;
-    doap:name "Airwindows BrassRider" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BrassRider@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BrassRider.ttl> .
-
-airwindows:brightambience
-    a lv2:Plugin ;
-    doap:name "Airwindows BrightAmbience" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BrightAmbience@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BrightAmbience.ttl> .
-
-airwindows:buildatpdf
-    a lv2:Plugin ;
-    doap:name "Airwindows BuildATPDF" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BuildATPDF@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BuildATPDF.ttl> .
-
-airwindows:busscolors4
-    a lv2:Plugin ;
-    doap:name "Airwindows BussColors4" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <BussColors4@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <BussColors4.ttl> .
-
-airwindows:buttercomp2
-    a lv2:Plugin ;
-    doap:name "Airwindows ButterComp2" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <ButterComp2@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <ButterComp2.ttl> .
-
-airwindows:buttercomp
-    a lv2:Plugin ;
-    doap:name "Airwindows ButterComp" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <ButterComp@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <ButterComp.ttl> .
-
-airwindows:c5rawbuss
-    a lv2:Plugin ;
-    doap:name "Airwindows C5RawBuss" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <C5RawBuss@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <C5RawBuss.ttl> .
-
-airwindows:c5rawchannel
-    a lv2:Plugin ;
-    doap:name "Airwindows C5RawChannel" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <C5RawChannel@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <C5RawChannel.ttl> .
-
-airwindows:calibre
-    a lv2:Plugin ;
-    doap:name "Airwindows Calibre" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Calibre@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Calibre.ttl> .
-
-airwindows:capacitor
-    a lv2:Plugin ;
-    doap:name "Airwindows Capacitor" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Capacitor@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Capacitor.ttl> .
-
-airwindows:channel4
-    a lv2:Plugin ;
-    doap:name "Airwindows Channel4" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Channel4@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Channel4.ttl> .
-
-airwindows:channel5
-    a lv2:Plugin ;
-    doap:name "Airwindows Channel5" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Channel5@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Channel5.ttl> .
-
-airwindows:channel6
-    a lv2:Plugin ;
-    doap:name "Airwindows Channel6" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Channel6@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Channel6.ttl> .
-
-airwindows:channel7
-    a lv2:Plugin ;
-    doap:name "Airwindows Channel7" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Channel7@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Channel7.ttl> .
-
-airwindows:chorus
-    a lv2:Plugin ;
-    doap:name "Airwindows Chorus" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Chorus@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Chorus.ttl> .
-
-airwindows:chorusensemble
-    a lv2:Plugin ;
-    doap:name "Airwindows ChorusEnsemble" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <ChorusEnsemble@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <ChorusEnsemble.ttl> .
-
-airwindows:chromeoxide
-    a lv2:Plugin ;
-    doap:name "Airwindows ChromeOxide" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <ChromeOxide@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <ChromeOxide.ttl> .
-
-airwindows:cider
-    a lv2:Plugin ;
-    doap:name "Airwindows Cider" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Cider@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Cider.ttl> .
-
-airwindows:cliponly
-    a lv2:Plugin ;
-    doap:name "Airwindows ClipOnly" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <ClipOnly@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <ClipOnly.ttl> .
-
-airwindows:coils
-    a lv2:Plugin ;
-    doap:name "Airwindows Coils" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Coils@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Coils.ttl> .
-
-airwindows:cojones
-    a lv2:Plugin ;
-    doap:name "Airwindows Cojones" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Cojones@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Cojones.ttl> .
-
-airwindows:compresaturator
-    a lv2:Plugin ;
-    doap:name "Airwindows Compresaturator" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Compresaturator@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Compresaturator.ttl> .
-
-airwindows:cstrip
-    a lv2:Plugin ;
-    doap:name "Airwindows CStrip" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <CStrip@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <CStrip.ttl> .
-
-airwindows:console5buss
-    a lv2:Plugin ;
-    doap:name "Airwindows Console5Buss" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Console5Buss@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Console5Buss.ttl> .
-
-airwindows:console5channel
-    a lv2:Plugin ;
-    doap:name "Airwindows Console5Channel" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Console5Channel@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Console5Channel.ttl> .
-
-airwindows:console5darkch
-    a lv2:Plugin ;
-    doap:name "Airwindows Console5DarkCh" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Console5DarkCh@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Console5DarkCh.ttl> .
-
-airwindows:console6buss
-    a lv2:Plugin ;
-    doap:name "Airwindows Console6Buss" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Console6Buss@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Console6Buss.ttl> .
-
-airwindows:console6channel
-    a lv2:Plugin ;
-    doap:name "Airwindows Console6Channel" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Console6Channel@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Console6Channel.ttl> .
-
-airwindows:crunchygroovewear
-    a lv2:Plugin ;
-    doap:name "Airwindows CrunchyGrooveWear" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <CrunchyGrooveWear@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <CrunchyGrooveWear.ttl> .
-
-airwindows:crystal
-    a lv2:Plugin ;
-    doap:name "Airwindows Crystal" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Crystal@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Crystal.ttl> .
-
-airwindows:curve
-    a lv2:Plugin ;
-    doap:name "Airwindows curve" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <curve@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <curve.ttl> .
-
-airwindows:dcvoltage
-    a lv2:Plugin ;
-    doap:name "Airwindows DCVoltage" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DCVoltage@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DCVoltage.ttl> .
-
-airwindows:dark
-    a lv2:Plugin ;
-    doap:name "Airwindows Dark" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Dark@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Dark.ttl> .
-
-airwindows:debess
-    a lv2:Plugin ;
-    doap:name "Airwindows DeBess" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DeBess@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DeBess.ttl> .
-
-airwindows:deess
-    a lv2:Plugin ;
-    doap:name "Airwindows DeEss" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DeEss@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DeEss.ttl> .
-
-airwindows:dehiss
-    a lv2:Plugin ;
-    doap:name "Airwindows DeHiss" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DeHiss@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DeHiss.ttl> .
-
-airwindows:derez
-    a lv2:Plugin ;
-    doap:name "Airwindows DeRez" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DeRez@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DeRez.ttl> .
-
-airwindows:derez2
-    a lv2:Plugin ;
-    doap:name "Airwindows DeRez2" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DeRez2@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DeRez2.ttl> .
-
-airwindows:deckwrecka
-    a lv2:Plugin ;
-    doap:name "Airwindows Deckwrecka" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Deckwrecka@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Deckwrecka.ttl> .
-
-airwindows:density
-    a lv2:Plugin ;
-    doap:name "Airwindows Density" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Density@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Density.ttl> .
-
-airwindows:desk
-    a lv2:Plugin ;
-    doap:name "Airwindows Desk" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Desk@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Desk.ttl> .
-
-airwindows:desk4
-    a lv2:Plugin ;
-    doap:name "Airwindows Desk4" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Desk4@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Desk4.ttl> .
-
-airwindows:digitalblack
-    a lv2:Plugin ;
-    doap:name "Airwindows DigitalBlack" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DigitalBlack@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DigitalBlack.ttl> .
-
-airwindows:distance
-    a lv2:Plugin ;
-    doap:name "Airwindows Distance" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Distance@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Distance.ttl> .
-
-airwindows:distance2
-    a lv2:Plugin ;
-    doap:name "Airwindows Distance2" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Distance2@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Distance2.ttl> .
-
-airwindows:distortion
-    a lv2:Plugin ;
-    doap:name "Airwindows Distortion" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Distortion@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Distortion.ttl> .
-
-airwindows:ditherfloat
-    a lv2:Plugin ;
-    doap:name "Airwindows DitherFloat" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DitherFloat@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DitherFloat.ttl> .
-
-airwindows:dithermediskers
-    a lv2:Plugin ;
-    doap:name "Airwindows DitherMeDiskers" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DitherMeDiskers@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DitherMeDiskers.ttl> .
-
-airwindows:dithermetimbers
-    a lv2:Plugin ;
-    doap:name "Airwindows DitherMeTimbers" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DitherMeTimbers@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DitherMeTimbers.ttl> .
-
-airwindows:ditherbox
-    a lv2:Plugin ;
-    doap:name "Airwindows Ditherbox" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Ditherbox@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Ditherbox.ttl> .
-
-airwindows:doublepaul
-    a lv2:Plugin ;
-    doap:name "Airwindows DoublePaul" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DoublePaul@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DoublePaul.ttl> .
-
-airwindows:drive
-    a lv2:Plugin ;
-    doap:name "Airwindows Drive" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Drive@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Drive.ttl> .
-
-airwindows:drumslam
-    a lv2:Plugin ;
-    doap:name "Airwindows DrumSlam" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DrumSlam@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DrumSlam.ttl> .
-
-airwindows:dubcenter
-    a lv2:Plugin ;
-    doap:name "Airwindows DubCenter" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DubCenter@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DubCenter.ttl> .
-
-airwindows:dubsub
-    a lv2:Plugin ;
-    doap:name "Airwindows DubSub" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DubSub@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DubSub.ttl> .
-
-airwindows:dustbunny
-    a lv2:Plugin ;
-    doap:name "Airwindows DustBunny" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <DustBunny@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <DustBunny.ttl> .
-
-airwindows:dyno
-    a lv2:Plugin ;
-    doap:name "Airwindows Dyno" ;
-    doap:license <http://opensource.org/licenses/mit> ;
-    lv2:binary <Dyno@CMAKE_SHARED_LIBRARY_SUFFIX@> ;
-    rdfs:seeAlso <Dyno.ttl> .
