diff --git a/Makefile b/Makefile
index f942741..c71a7f3 100644
--- a/Makefile
+++ b/Makefile
@@ -80,9 +80,9 @@ src/mod-monitor.o: src/monitor/monitor-client.c
 install: install_man
 	install -d $(DESTDIR)$(BINDIR)
 	install -m 755 $(PROG) $(DESTDIR)$(BINDIR)
-	install -d $(DESTDIR)$(shell pkg-config --variable=libdir jack)/jack/
-	install -m 755 $(PROG).so $(DESTDIR)$(shell pkg-config --variable=libdir jack)/jack/
-	install -m 755 mod-monitor.so $(DESTDIR)$(shell pkg-config --variable=libdir jack)/jack/
+	install -d $(DESTDIR)/usr/lib/jack/
+	install -m 755 $(PROG).so $(DESTDIR)/usr/lib/jack/
+	install -m 755 mod-monitor.so $(DESTDIR)/usr/lib/jack/
 
 # clean rule
 clean:
