diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index 8915fba0697a..293f11b6049b 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -1281,12 +1281,20 @@ static int irq_thread(void *data)
 	struct callback_head on_exit_work;
 	struct irqaction *action = data;
 	struct irq_desc *desc = irq_to_desc(action->irq);
+	struct sched_param sp = { .sched_priority = MAX_RT_PRIO / 2 };
 	irqreturn_t (*handler_fn)(struct irq_desc *desc,
 			struct irqaction *action);
 
+	/**/ if (!strcmp(action->name, "ff240000.dma-controller"))
+		sp.sched_priority = 91;
+	else if (!strcmp(action->name, "ff1a0000.i2c"))
+		sp.sched_priority = 74;
+	else if (!strcmp(action->name, "ff300000.usb") || !strcmp(action->name, "dwc2_hsotg:usb3"))
+		sp.sched_priority = 72;
+
 	irq_thread_set_ready(desc, action);
 
-	sched_set_fifo(current);
+	sched_setscheduler_nocheck(current, SCHED_FIFO, &sp);
 
 	if (force_irqthreads() && test_bit(IRQTF_FORCED_THREAD,
 					   &action->thread_flags))
