diff --git a/ports-legacy/meson.build b/ports-legacy/meson.build
index cc8a600..abb447e 100644
--- a/ports-legacy/meson.build
+++ b/ports-legacy/meson.build
@@ -280,8 +278,7 @@ if build_lv2 or build_vst2
                     command: [
                         'mkdir', '-p', plugin_lv2_dir, '&&',
                         'cd', plugin_lv2_dir, '&&',
-                        'cp', plugin_lv2_lib.full_path(), plugin_lv2_dir / plugin_name + lib_suffix, '&&',
-                        (meson.is_cross_build() ? 'wine' : 'env'), lv2_ttl_generator, '.' / plugin_name + lib_suffix,
+                        'mv', plugin_lv2_lib.full_path(), plugin_lv2_dir / plugin_name + lib_suffix,
                     ],
                     install: true,
                     install_dir: lv2dir,
diff --git a/ports/meson.build b/ports/meson.build
index 6369385..ecdf549 100644
--- a/ports/meson.build
+++ b/ports/meson.build
@@ -225,8 +223,7 @@ foreach plugin : plugins
                 command: [
                     'mkdir', '-p', plugin_lv2_dir, '&&',
                     'cd', plugin_lv2_dir, '&&',
-                    'mv', plugin_lv2_lib.full_path(), plugin_lv2_dir / plugin_name + lib_suffix, '&&',
-                    (meson.is_cross_build() ? 'wine' : 'env'), lv2_ttl_generator, '.' / plugin_name + lib_suffix,
+                    'mv', plugin_lv2_lib.full_path(), plugin_lv2_dir / plugin_name + lib_suffix,
                 ],
                 install: true,
                 install_dir: lv2dir,
diff --git a/scripts/plugin-symbols-lv2.version b/scripts/plugin-symbols-lv2.version
index 8f21307..328c6f9 100644
--- a/scripts/plugin-symbols-lv2.version
+++ b/scripts/plugin-symbols-lv2.version
@@ -1,4 +1,4 @@
 {
-    global: lv2_descriptor; lv2ui_descriptor; lv2_generate_ttl;
+    global: lv2_descriptor; lv2ui_descriptor;
     local: *;
 };
