--- abgate-1.1.7.orig/Makefile
+++ abgate-1.1.7/Makefile
@@ -13,10 +13,10 @@ abGateQt/libabGateQt.so:
 	cd abGateQt; qmake-qt4; make
 
 gate.so: gate.cpp
-	g++ $(LDFLAGS) -O3 -shared -fPIC -DPIC gate.cpp `pkg-config --cflags --libs lv2` -o gate.so
+	g++ $(CXXFLAGS) $(LDFLAGS) -O3 -shared -fPIC -DPIC gate.cpp `pkg-config --cflags --libs lv2` -o gate.so
 
 gate_gui.so: gate_gui.cpp main_window.cpp main_window.h knob.cpp knob.h toggle.cpp toggle.h preset_widget.cpp preset_widget.h presets.cpp presets.h preset.cpp preset.h gate_const.h ui.h plugin_configuration.h
-	g++ $(LDFLAGS) -O3 -shared -fPIC -DPIC gate_gui.cpp main_window.cpp knob.cpp toggle.cpp preset_widget.cpp presets.cpp preset.cpp `pkg-config --cflags gtkmm-2.4 --libs lv2 gthread-2.0` -o gate_gui.so
+	g++ $(CXXFLAGS) $(LDFLAGS) -O3 -shared -fPIC -DPIC gate_gui.cpp main_window.cpp knob.cpp toggle.cpp preset_widget.cpp presets.cpp preset.cpp `pkg-config --cflags gtkmm-2.4 --libs lv2 gthread-2.0` -o gate_gui.so
 
 install: $(BUNDLE)
 	mkdir -p $(INSTALL_DIR)
