From fbe77164cf58807807b8c6ec496430f34f88d780 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@falktx.com>
Date: Mon, 12 Oct 2020 12:00:01 +0100
Subject: [PATCH] Fix build against latest LV2

Signed-off-by: falkTX <falktx@falktx.com>
---
 gate.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gate.cpp b/gate.cpp
index a09d2f6..bfd3135 100644
--- a/gate.cpp
+++ b/gate.cpp
@@ -44,7 +44,7 @@ class Gate {
 	int state, holding;
 };
 
-static LV2_Handle instantiateGate(const _LV2_Descriptor *descriptor, double s_rate, const char *path, const LV2_Feature * const* features) {
+static LV2_Handle instantiateGate(const LV2_Descriptor *descriptor, double s_rate, const char *path, const LV2_Feature * const * features) {
 	Gate *plugin_data = new Gate;
 	plugin_data->sample_rate = s_rate;
 	return (LV2_Handle)plugin_data;
