diff --git a/src/core/LSPString.cpp b/src/core/LSPString.cpp
index 38dc057..8266897 100644
--- a/src/core/LSPString.cpp
+++ b/src/core/LSPString.cpp
@@ -1655,7 +1655,7 @@ namespace lsp
         LSPString temp;
 
         // Open conversion
-        iconv_t cd = init_iconv_to_wchar_t(charset);
+        iconv_t cd = iconv_open(charset ? charset : "UTF-8", "WCHAR_T");
         if (cd == iconv_t(-1))
             return false;
 
@@ -1879,7 +1879,7 @@ namespace lsp
             return (last == first) ? "" : NULL;
 
         // Open conversion
-        iconv_t cd = init_iconv_from_wchar_t(charset);
+        iconv_t cd = iconv_open(charset ? charset : "UTF-8", "WCHAR_T");
         if (cd == iconv_t(-1))
             return NULL;
 
