diff --git a/source/backend/engine/CarlaEngineDummy.cpp b/source/backend/engine/CarlaEngineDummy.cpp
index 4477776ef..9bf129c05 100644
--- a/source/backend/engine/CarlaEngineDummy.cpp
+++ b/source/backend/engine/CarlaEngineDummy.cpp
@@ -69,8 +69,8 @@ public:
             return false;
         }
 
-        pData->bufferSize = pData->options.audioBufferSize;
-        pData->sampleRate = pData->options.audioSampleRate;
+        pData->bufferSize = 256;
+        pData->sampleRate = 48000;
         pData->initTime(pData->options.transportExtra);
 
         pData->graph.create(2, 2, 0, 0);
