diff --git a/libmount/src/hook_mount.c b/libmount/src/hook_mount.c
index 056338c..9df3bf2 100644
--- a/libmount/src/hook_mount.c
+++ b/libmount/src/hook_mount.c
@@ -294,7 +294,7 @@ static int hook_create_mount(struct libmnt_context *cxt,
 		/* cleanup after fail (libmount may only try the FS type) */
 		close_sysapi_fds(api);
 
-#if defined(HAVE_STRUCT_STATX) && defined(HAVE_STRUCT_STATX_STX_MNT_ID)
+#if 0
 	if (!rc && cxt->fs) {
 		struct statx st;
 
