diff --git a/screcord/Makefile b/screcord/Makefile
index 584d094..95383dc 100644
--- a/screcord/Makefile
+++ b/screcord/Makefile
@@ -62,7 +62,7 @@
 
 	# set compile flags
 	CFLAGS += -I. -I./dsp  -fPIC -DPIC -O2 -Wall -funroll-loops `pkg-config --cflags sndfile`\
-	-ffast-math -fomit-frame-pointer -fstrength-reduce -fdata-sections -Wl,--gc-sections $(SSE_CFLAGS)
+	-ffast-math -fomit-frame-pointer -fstrength-reduce -fdata-sections -Wl,--gc-sections -pthread $(SSE_CFLAGS)
 	CXXFLAGS += -std=c++11 $(CFLAGS)
 	LDFLAGS += -I. -lm -shared -Llibrary -lm -fPIC -DPIC `pkg-config --libs sndfile`
 	ifneq ($(MACOS)$(WINDOWS),true)
