diff --git a/screcord/Makefile b/screcord/Makefile
index 959ccbf..d3b2f6d 100644
--- a/screcord/Makefile
+++ b/screcord/Makefile
@@ -63,7 +63,10 @@
 	# set compile flags
 	CXXFLAGS += -I. -I./dsp  -fPIC -DPIC -O2 -Wall -funroll-loops `pkg-config --cflags sndfile`\
 	-ffast-math -fomit-frame-pointer -fstrength-reduce -fdata-sections -Wl,--gc-sections $(SSE_CFLAGS)
-	LDFLAGS += -I.  -lm -pthread -shared -Llibrary -lc -lm -lrt -fPIC -DPIC `pkg-config --libs sndfile` 
+	LDFLAGS += -I.  -lm -pthread -shared -Llibrary -lm -fPIC -DPIC `pkg-config --libs sndfile`
+ifneq ($(MACOS)$(WINDOWS),true)
+	LDFLAGS += -lrt
+endif
 	GUI_LDFLAGS += -I. -I$(HEADER_DIR) \
 	-L. $(LIB_DIR)libxputty.a -shared `pkg-config --static --cflags --libs cairo x11` -lm 
 	# invoke build files
diff --git a/screcord/screcord1.cc b/screcord/screcord1.cc
index 19894da..2ac4a5c 100644
--- a/screcord/screcord1.cc
+++ b/screcord/screcord1.cc
@@ -146,7 +146,11 @@ inline std::string SCapture::get_ffilename() {
 #endif
     is_wav = int(*fformat) ? false : true;
     if (!(stat(pPath.c_str(), &sb) == 0 && S_ISDIR(sb.st_mode))) {
+#ifdef _WIN32
+        mkdir(pPath.c_str());
+#else
         mkdir(pPath.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+#endif
     }
 
 #ifndef  __MOD_DEVICES__
