diff --git a/linux-user/elfload.c b/linux-user/elfload.c
index a69e7d7..6e0dc35 100644
--- a/linux-user/elfload.c
+++ b/linux-user/elfload.c
@@ -2519,7 +2519,7 @@ static int pgb_try_mmap(uintptr_t addr, uintptr_t addr_last, bool keep)
     size_t size = addr_last - addr + 1;
     void *p = mmap((void *)addr, size, PROT_NONE,
                    MAP_ANONYMOUS | MAP_PRIVATE |
-                   MAP_NORESERVE | MAP_FIXED_NOREPLACE, -1, 0);
+                   MAP_NORESERVE | 0x100000, -1, 0);
     int ret;
 
     if (p == MAP_FAILED) {
@@ -3168,7 +3168,7 @@ static void load_elf_image(const char *image_name, int image_fd,
      */
     load_addr = target_mmap(load_addr, (size_t)hiaddr - loaddr + 1, PROT_NONE,
                             MAP_PRIVATE | MAP_ANON | MAP_NORESERVE |
-                            (ehdr->e_type == ET_EXEC ? MAP_FIXED_NOREPLACE : 0),
+                            (ehdr->e_type == ET_EXEC ? 0x100000 : 0),
                             -1, 0);
     if (load_addr == -1) {
         goto exit_mmap;
diff --git a/linux-user/mmap.c b/linux-user/mmap.c
index 9aab48d..95074c7 100644
--- a/linux-user/mmap.c
+++ b/linux-user/mmap.c
@@ -500,7 +500,7 @@ abi_long target_mmap(abi_ulong start, abi_ulong len, int target_prot,
      * If the user is asking for the kernel to find a location, do that
      * before we truncate the length for mapping files below.
      */
-    if (!(flags & (MAP_FIXED | MAP_FIXED_NOREPLACE))) {
+    if (!(flags & (MAP_FIXED | 0x100000))) {
         host_len = len + offset - host_offset;
         host_len = HOST_PAGE_ALIGN(host_len);
         start = mmap_find_vma(real_start, host_len, TARGET_PAGE_SIZE);
@@ -542,7 +542,7 @@ abi_long target_mmap(abi_ulong start, abi_ulong len, int target_prot,
         }
     }
 
-    if (!(flags & (MAP_FIXED | MAP_FIXED_NOREPLACE))) {
+    if (!(flags & (MAP_FIXED | 0x100000))) {
         uintptr_t host_start;
         int host_prot;
         void *p;
@@ -594,7 +594,7 @@ abi_long target_mmap(abi_ulong start, abi_ulong len, int target_prot,
             goto fail;
         }
 
-        if (flags & MAP_FIXED_NOREPLACE) {
+        if (flags & 0x100000) {
             /* Validate that the chosen range is empty. */
             if (!page_check_range_empty(start, last)) {
                 errno = EEXIST;
@@ -612,7 +612,7 @@ abi_long target_mmap(abi_ulong start, abi_ulong len, int target_prot,
              * any host address mappings.
              */
             if (reserved_va) {
-                flags = (flags & ~MAP_FIXED_NOREPLACE) | MAP_FIXED;
+                flags = (flags & ~0x100000) | MAP_FIXED;
             }
         }
 
@@ -632,7 +632,7 @@ abi_long target_mmap(abi_ulong start, abi_ulong len, int target_prot,
                 goto fail;
             }
             retaddr = target_mmap(start, len, target_prot | PROT_WRITE,
-                                  (flags & (MAP_FIXED | MAP_FIXED_NOREPLACE))
+                                  (flags & (MAP_FIXED | 0x100000))
                                   | MAP_PRIVATE | MAP_ANONYMOUS,
                                   -1, 0);
             if (retaddr == -1) {
diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 9353268..6736e28 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -841,7 +841,7 @@ abi_long do_brk(abi_ulong brk_val)
 
     mapped_addr = target_mmap(old_brk, new_brk - old_brk,
                               PROT_READ | PROT_WRITE,
-                              MAP_FIXED_NOREPLACE | MAP_ANON | MAP_PRIVATE,
+                              0x100000 | MAP_ANON | MAP_PRIVATE,
                               -1, 0);
 
     if (mapped_addr == old_brk) {
@@ -6000,7 +6000,7 @@ static const bitmask_transtbl mmap_flags_tbl[] = {
     { TARGET_MAP_NONBLOCK, TARGET_MAP_NONBLOCK, MAP_NONBLOCK, MAP_NONBLOCK },
     { TARGET_MAP_POPULATE, TARGET_MAP_POPULATE, MAP_POPULATE, MAP_POPULATE },
     { TARGET_MAP_FIXED_NOREPLACE, TARGET_MAP_FIXED_NOREPLACE,
-      MAP_FIXED_NOREPLACE, MAP_FIXED_NOREPLACE },
+      0x100000, 0x100000 },
     { TARGET_MAP_UNINITIALIZED, TARGET_MAP_UNINITIALIZED,
       MAP_UNINITIALIZED, MAP_UNINITIALIZED },
 };
@@ -6061,7 +6061,7 @@ static abi_long do_mmap(abi_ulong addr, abi_ulong len, int prot,
         if (target_flags & ~(TARGET_LEGACY_MAP_MASK | TARGET_MAP_SYNC)) {
             return -TARGET_EOPNOTSUPP;
         }
-        host_flags = MAP_SHARED_VALIDATE;
+        host_flags = 0x03;
         if (target_flags & TARGET_MAP_SYNC) {
             host_flags |= MAP_SYNC;
         }
