diff --git a/src/Containers/NotePool.cpp b/src/Containers/NotePool.cpp
index d67d9141..74408fb5 100644
--- a/src/Containers/NotePool.cpp
+++ b/src/Containers/NotePool.cpp
@@ -155,7 +155,7 @@ void NotePool::insertNote(note_t note, uint8_t sendto, SynthDescriptor desc, boo
 {
     //Get first free note descriptor
     int desc_id = getMergeableDescriptor(note, sendto, legato, ndesc);
-    assert(desc_id != -1);
+    if (desc_id == -1) return;
 
     ndesc[desc_id].note         = note;
     ndesc[desc_id].sendto       = sendto;
