diff --git a/src/globals.h b/src/globals.h
index 01918fa1..13fb407b 100644
--- a/src/globals.h
+++ b/src/globals.h
@@ -165,7 +165,7 @@ typedef std::complex<fftwf_real> fft_t;
 /*
  * Maximum filter stages
  */
-#define MAX_FILTER_STAGES 5
+#define MAX_FILTER_STAGES 5 // FIXME ensure remote UI can work when changing this
 
 /*
  * Formant filter (FF) limits
@@ -294,7 +294,7 @@ public:
 struct SYNTH_T {
 
     SYNTH_T(void)
-        :samplerate(44100), buffersize(256), oscilsize(1024)
+        :samplerate(48000), buffersize(128), oscilsize(512)
     {
         alias(false);
     }
