diff --git a/common/JackNetManager.cpp b/common/JackNetManager.cpp
index 250c123..b09fcd7 100644
--- a/common/JackNetManager.cpp
+++ b/common/JackNetManager.cpp
@@ -186,15 +186,15 @@ namespace Jack
 
         //audio
         for (i = 0; i < fParams.fSendAudioChannels; i++) {
-            snprintf(name, sizeof(name), "to_slave_%d", i+1);
-            if ((fAudioCapturePorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_AUDIO_TYPE, JackPortIsInput | JackPortIsTerminal, 0)) == NULL) {
+            snprintf(name, sizeof(name), "audio_to_slave_%d", i+1);
+            if ((fAudioCapturePorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_AUDIO_TYPE, JackPortIsInput | JackPortIsTerminal | JackPortIsPhysical, 0)) == NULL) {
                 return -1;
             }
         }
 
         for (i = 0; i < fParams.fReturnAudioChannels; i++) {
-            snprintf(name, sizeof(name), "from_slave_%d", i+1);
-            if ((fAudioPlaybackPorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_AUDIO_TYPE, JackPortIsOutput | JackPortIsTerminal, 0)) == NULL) {
+            snprintf(name, sizeof(name), "audio_from_slave_%d", i+1);
+            if ((fAudioPlaybackPorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_AUDIO_TYPE, JackPortIsOutput | JackPortIsTerminal | JackPortIsPhysical, 0)) == NULL) {
                 return -1;
             }
         }
@@ -202,14 +202,14 @@ namespace Jack
         //midi
         for (i = 0; i < fParams.fSendMidiChannels; i++) {
             snprintf(name, sizeof(name), "midi_to_slave_%d", i+1);
-            if ((fMidiCapturePorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_MIDI_TYPE, JackPortIsInput | JackPortIsTerminal, 0)) == NULL) {
+            if ((fMidiCapturePorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_MIDI_TYPE, JackPortIsInput | JackPortIsTerminal | JackPortIsPhysical, 0)) == NULL) {
                 return -1;
             }
         }
 
         for (i = 0; i < fParams.fReturnMidiChannels; i++) {
             snprintf(name, sizeof(name), "midi_from_slave_%d", i+1);
-            if ((fMidiPlaybackPorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_MIDI_TYPE,  JackPortIsOutput | JackPortIsTerminal, 0)) == NULL) {
+            if ((fMidiPlaybackPorts[i] = jack_port_register(fClient, name, JACK_DEFAULT_MIDI_TYPE,  JackPortIsOutput | JackPortIsTerminal | JackPortIsPhysical, 0)) == NULL) {
                 return -1;
             }
         }
