diff --git a/scripts/make/tools.mk b/scripts/make/tools.mk
index bd6e01f..3be175a 100644
--- a/scripts/make/tools.mk
+++ b/scripts/make/tools.mk
@@ -27,7 +27,7 @@ LD                       ?= $(TOOL_LD)
 MAKE_OPTS                 = -s
 CFLAGS                   += $(CC_ARCH) $(FLAG_CTUNE) $(CC_FLAGS) $(FLAG_VERSION)
 CXXFLAGS                 += $(CC_ARCH) $(FLAG_CTUNE) $(CC_FLAGS) $(FLAG_VERSION)
-SO_FLAGS                  = $(CC_ARCH) $(FLAG_RELRO) -Wl,--gc-sections -shared -Llibrary -lc -fPIC
+SO_FLAGS                  = $(CC_ARCH) $(FLAG_RELRO) -Wl,--gc-sections -Wl,--no-undefined -shared -Llibrary -lc -fPIC
 MERGE_FLAGS               = $(LD_ARCH) -r
 EXE_TEST_FLAGS            = $(LDFLAGS) $(CC_ARCH)
 EXE_FLAGS                 = $(LDFLAGS) $(CC_ARCH) $(FLAG_RELRO) -Wl,--gc-sections
diff --git a/src/container/lv2.cpp b/src/container/lv2.cpp
index 4b94aeb..b00b15b 100644
--- a/src/container/lv2.cpp
+++ b/src/container/lv2.cpp
@@ -11,15 +11,19 @@
 
 #include <data/cvector.h>
 
+#ifndef __MOD_DEVICES__
 #include <ui/ui.h>
+#endif
 
 #include <container/lv2/extensions.h>
 #include <container/lv2/types.h>
 #include <container/lv2/ports.h>
 #include <container/lv2/executor.h>
 #include <container/lv2/wrapper.h>
+#ifndef __MOD_DEVICES__
 #include <container/lv2/ui_ports.h>
 #include <container/lv2/ui_wrapper.h>
+#endif
 
 namespace lsp
 {
@@ -279,6 +283,7 @@ namespace lsp
 
     static StaticFinalizer lv2_finalizer(lv2_drop_descriptors);
 
+#ifndef __MOD_DEVICES__
     //--------------------------------------------------------------------------------------
     // LV2UI routines
 
@@ -418,6 +423,7 @@ namespace lsp
     };
 
     static StaticFinalizer lv2ui_finalizer(lv2ui_drop_descriptors);
+#endif
 }
 
 #ifdef __cplusplus
@@ -435,6 +441,7 @@ extern "C"
         return (index < lv2_descriptors_count) ? &lv2_descriptors[index] : NULL;
     }
 
+#ifndef __MOD_DEVICES__
     LV2_SYMBOL_EXPORT
     const LV2UI_Descriptor *lv2ui_descriptor(uint32_t index)
     {
@@ -444,6 +451,7 @@ extern "C"
         lv2ui_gen_descriptors();
         return (index < lv2ui_descriptors_count) ? &lv2ui_descriptors[index] : NULL;
     }
+#endif
 
 #ifdef __cplusplus
 }
