diff --git a/kernel/ifftw.h b/kernel/ifftw.h
index 810ce8a..754772d 100644
--- a/kernel/ifftw.h
+++ b/kernel/ifftw.h
@@ -103,6 +103,7 @@ extern void X(extract_reim)(int sign, R *c, R **r, R **i);
       defined(HAVE_KCVI) || \
       defined(HAVE_ALTIVEC) || defined(HAVE_VSX) || \
       defined(HAVE_MIPS_PS) || \
+      defined(HAVE_NEON) || \
       defined(HAVE_GENERIC_SIMD128) || defined(HAVE_GENERIC_SIMD256)
 #define HAVE_SIMD 1
 #else
