diff --git a/kernel/planner.c b/kernel/planner.c
index 172c398..4dabb04 100644
--- a/kernel/planner.c
+++ b/kernel/planner.c
@@ -20,6 +20,7 @@
 
 #include "ifftw.h"
 #include <string.h>
+#include <stdio.h>
 
 /* GNU Coding Standards, Sec. 5.2: "Please write the comments in a GNU
    program in English, because English is the one language that nearly
@@ -847,6 +848,7 @@ static int imprt(planner *ego, scanner *sc)
      signature_of_configuration(&m, ego);
      if (m.s[0] != sig[0] || m.s[1] != sig[1] ||
 	 m.s[2] != sig[2] || m.s[3] != sig[3]) {
+	  printf("Wisdom config mismatch, wanted #x%08x #x%08x #x%08x #x%08x\n", m.s[0], m.s[1], m.s[2], m.s[3]);
 	  /* invalid configuration */
 	  return 0;
      }
