diff --git a/Makefile b/Makefile
index 56eba01..55c3244 100644
--- a/Makefile
+++ b/Makefile
@@ -109,9 +109,7 @@ lv2:
 	$(MAKE) gen_lv2
 
 lv2_nogen:
-	$(MAKE) -C libs/drowaudio
 	$(MAKE) -C libs/juce
-	$(MAKE) -C libs/juced
 	$(MAKE) -C ports lv2
 
 vst:
diff --git a/ports/Makefile b/ports/Makefile
index 249f79f..23f8299 100644
--- a/ports/Makefile
+++ b/ports/Makefile
@@ -20,26 +20,7 @@ libs:
 lv2: libs
 	install -d ../bin/lv2
 ifeq ($(LINUX_EMBED),true)
-	$(MAKE) -C arctican-function/LV2
-	$(MAKE) -C arctican-pilgrim/LV2
-	$(MAKE) -C drowaudio-distortion/LV2
-	$(MAKE) -C drowaudio-distortionshaper/LV2
-	$(MAKE) -C drowaudio-flanger/LV2
-	$(MAKE) -C drowaudio-reverb/LV2
-	$(MAKE) -C drowaudio-tremolo/LV2
-	$(MAKE) -C luftikus/LV2
-	$(MAKE) -C obxd/LV2
-	$(MAKE) -C tal-dub-3/LV2
-	$(MAKE) -C tal-filter/LV2
-	$(MAKE) -C tal-filter-2/LV2
-	$(MAKE) -C tal-noisemaker/LV2
-	$(MAKE) -C tal-reverb/LV2
-	$(MAKE) -C tal-reverb-2/LV2
-	$(MAKE) -C tal-reverb-3/LV2
-	$(MAKE) -C tal-vocoder-2/LV2
-	$(MAKE) -C temper/LV2
 	$(MAKE) -C vex/LV2
-	$(MAKE) -C wolpertinger/LV2
 else
 	$(MAKE) -C arctican-function/LV2
 	$(MAKE) -C arctican-pilgrim/LV2
