From aae90ba38f099e7de1e0bb2de868dcfefdf8687b Mon Sep 17 00:00:00 2001
From: falkTX <falktx@falktx.com>
Date: Wed, 20 Apr 2022 13:42:26 +0100
Subject: [PATCH] Fix lv2 syntax validation

Signed-off-by: falkTX <falktx@falktx.com>
---
 resources/aether.ttl.in | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/resources/aether.ttl.in b/resources/aether.ttl.in
index 2f79999..db9089b 100644
--- a/resources/aether.ttl.in
+++ b/resources/aether.ttl.in
@@ -19,15 +19,17 @@
 	foaf:homepage <http://dougal-s.github.io>.
 
 <http://dougal-s.github.io#plugins>
-	a doap:project;
+	a doap:Project;
 	doap:maintainer <http://dougal-s.github.io>;
 	doap:name "Aether".
 
 <http://github.com/Dougal-s/Aether#input>
-	a pg:StereoGroup, pg:InputGroup.
+	a pg:StereoGroup, pg:InputGroup;
+	lv2:symbol "input".
 
 <http://github.com/Dougal-s/Aether#output>
 	a pg:StereoGroup, pg:OutputGroup;
+	lv2:symbol "output";
 	pg:source <http://github.com/Dougal-s/Aether#input>.
 
 <http://github.com/Dougal-s/Aether>
@@ -324,7 +326,7 @@
 		lv2:name "Late Order";
 		rdfs:comment "Controls the signal path in the late reverberation unit";
 		lv2:portProperty lv2:integer, lv2:enumeration;
-		lv2:scalePoints
+		lv2:scalePoint
 			[ rdfs:label "Pre"; rdf:value 0 ],
 			[ rdfs:label "Post"; rdf:value 1 ];
 		lv2:default 0;
@@ -597,7 +599,7 @@
 		lv2:default -12.0;
 		lv2:minimum -12.0;
 		lv2:maximum 12.0;
-		rdfs:scalePoints [
+		lv2:scalePoint [
 			rdfs:label "Off";
 			rdf:value -12.0
 		];
