diff --git a/src/casynth/CMakeLists.txt b/src/casynth/CMakeLists.txt
index 9c6a560..4e506b0 100644
--- a/src/casynth/CMakeLists.txt
+++ b/src/casynth/CMakeLists.txt
@@ -2,8 +2,8 @@
 
 set(PLUGIN "casynth")
 
-#set(CMAKE_C_FLAGS "-Wall -g -std=c99 -msse2 -mfpmath=sse -ffast-math")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -std=c99 -msse2 -mfpmath=sse -ffast-math")
+#set(CMAKE_C_FLAGS "-Wall -std=c99")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
diff --git a/src/cheapdist/CMakeLists.txt b/src/cheapdist/CMakeLists.txt
index b5bebe5..48eee9d 100644
--- a/src/cheapdist/CMakeLists.txt
+++ b/src/cheapdist/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(PLUGIN "cheapdist")
 
-set(CMAKE_C_FLAGS "-Wall -g -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_C_FLAGS "-Wall")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
diff --git a/src/duffer/CMakeLists.txt b/src/duffer/CMakeLists.txt
index 32f796c..b06c1ba 100644
--- a/src/duffer/CMakeLists.txt
+++ b/src/duffer/CMakeLists.txt
@@ -2,8 +2,8 @@
 
 set(PLUGIN "duffer")
 
-#set(CMAKE_C_FLAGS "-Wall -g -std=c99 -msse2 -mfpmath=sse -ffast-math")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -std=c99 -msse2 -mfpmath=sse -ffast-math")
+#set(CMAKE_C_FLAGS "-Wall -std=c99")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
diff --git a/src/envfollower/CMakeLists.txt b/src/envfollower/CMakeLists.txt
index 831493f..c39f867 100644
--- a/src/envfollower/CMakeLists.txt
+++ b/src/envfollower/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(PLUGIN "envfollower")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -std=c99 -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
diff --git a/src/ewham/CMakeLists.txt b/src/ewham/CMakeLists.txt
index 06dd7e5..6a1d8cc 100644
--- a/src/ewham/CMakeLists.txt
+++ b/src/ewham/CMakeLists.txt
@@ -4,7 +4,7 @@ set(PLUGIN "ewham")
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../cmake/")
 
-set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -Wall -g -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -Wall")
 #SET( CMAKE_SHARED_LINKER_FLAGS  "${CMAKE_SHARED_LINKER_FLAGS} -lzita-resampler")
 
 if(NOT LIBDIR)
diff --git a/src/hip2b/CMakeLists.txt b/src/hip2b/CMakeLists.txt
index 68f6e54..f1620c8 100644
--- a/src/hip2b/CMakeLists.txt
+++ b/src/hip2b/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(PLUGIN "hip2b")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -g -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
diff --git a/src/lushlife/CMakeLists.txt b/src/lushlife/CMakeLists.txt
index ddc392a..a5cdaf6 100644
--- a/src/lushlife/CMakeLists.txt
+++ b/src/lushlife/CMakeLists.txt
@@ -4,7 +4,7 @@ set(PLUGIN "lushlife")
 
 set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../../cmake/")
 
-set(CMAKE_CXX_FLAGS "-Wall -g -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_CXX_FLAGS "-Wall")
 #SET( CMAKE_SHARED_LINKER_FLAGS  "${CMAKE_SHARED_LINKER_FLAGS} -lzita-resampler")
 
 if(NOT LIBDIR)
diff --git a/src/powercut/CMakeLists.txt b/src/powercut/CMakeLists.txt
index a82cc61..bed64a7 100644
--- a/src/powercut/CMakeLists.txt
+++ b/src/powercut/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(PLUGIN "powercut")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -g -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
diff --git a/src/powerup/CMakeLists.txt b/src/powerup/CMakeLists.txt
index 506ac06..981ae73 100644
--- a/src/powerup/CMakeLists.txt
+++ b/src/powerup/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(PLUGIN "powerup")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -g -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
diff --git a/src/stuck/CMakeLists.txt b/src/stuck/CMakeLists.txt
index 4d23d0a..52b3c42 100644
--- a/src/stuck/CMakeLists.txt
+++ b/src/stuck/CMakeLists.txt
@@ -2,7 +2,7 @@
 
 set(PLUGIN "stuck")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -g -msse2 -mfpmath=sse -ffast-math")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
 
 if(NOT LIBDIR)
   set(LIBDIR "lib")
