diff --git a/include/resources/faust.h b/include/resources/faust.h
index 819e284..64a10dc 100644
--- a/include/resources/faust.h
+++ b/include/resources/faust.h
@@ -31,6 +31,8 @@
 #define __rt_data
 #define always_inline   inline __attribute__((__always_inline__))
 
+using std::signbit;
+
 namespace faust
 {
     /**
diff --git a/src/Makefile b/src/Makefile
index 6180851..848b424 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -55,6 +55,6 @@ $(GENERATED):
 
 $(OBJFILES): $(patsubst $(BUILD_DIR)/%.o, %.cpp, $(@))
 	mkdir -p $(dir $(@))
-	$(CXX) -o $(@) -c $(CPP_FILE) $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE)
+	$(CXX) -o $(@) -c $(CPP_FILE) -fPIC $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE)
 
 -include Makefile.d
diff --git a/src/resources/Makefile b/src/resources/Makefile
index 88c3caf..71dfb1e 100644
--- a/src/resources/Makefile
+++ b/src/resources/Makefile
@@ -37,6 +37,6 @@ $(BUILD_FILE): $(OBJFILES)
 
 $(OBJFILES):
 	mkdir -p $(dir $(@))
-	$(CXX) -o $(@) -c $(CPP_FILE) $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE)
+	$(CXX) -o $(@) -c $(CPP_FILE) -fPIC $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE)
 
 -include Makefile.d
diff --git a/src/tamgamp.lv2/Makefile b/src/tamgamp.lv2/Makefile
index 7bfa601..b333f31 100644
--- a/src/tamgamp.lv2/Makefile
+++ b/src/tamgamp.lv2/Makefile
@@ -37,6 +37,6 @@ $(BUILD_FILE): $(OBJFILES)
 
 $(OBJFILES):
 	mkdir -p $(dir $(@))
-	$(CXX) -o $(@) -c $(CPP_FILE) $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE)
+	$(CXX) -o $(@) -c $(CPP_FILE) -fPIC $(CPPFLAGS) $(CXXFLAGS) $(INCLUDE)
 
 -include Makefile.d
