--- a/mod-ams.lv2/env.ttl
+++ b/mod-ams.lv2/env.ttl
@@ -4,11 +4,6 @@
 @prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 @prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
 @prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/env/gui>
-  	a guiext:GtkUI ;
-  	guiext:binary <env_gui.so> .
 
 <http://github.com/blablack/ams-lv2/env>
 	a lv2:Plugin, lv2:EnvelopePlugin ;
@@ -16,7 +11,6 @@
 	doap:name "AMS Envelope" ;
 	doap:license <http://opensource.org/licenses/isc> ;
 	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/env/gui> ;
 	lv2:optionalFeature lv2:hardRTCapable ;
 	lv2:port [
 		a lv2:InputPort, lv2:CVPort ;

diff --git a/mod-ams.lv2/lfo2_freq.ttl b/mod-ams.lv2/lfo2_freq.ttl
index cc6611f..164220f 100644
--- a/mod-ams.lv2/lfo2_freq.ttl
+++ b/mod-ams.lv2/lfo2_freq.ttl
@@ -4,11 +4,6 @@
 @prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 @prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
 @prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/lfo2_freq/gui>
-  a guiext:GtkUI ;
-  guiext:binary <lfo2_freq_gui.so> .
 
 <http://github.com/blablack/ams-lv2/lfo2_freq>
 	a lv2:Plugin, lv2:OscillatorPlugin ;
@@ -16,7 +11,6 @@
 	doap:name "AMS LFO2 - Freq" ;
 	doap:license <http://opensource.org/licenses/isc> ;
 	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/lfo2_freq/gui> ;
 	lv2:optionalFeature lv2:hardRTCapable ;
 	lv2:port [
 		a lv2:InputPort, lv2:CVPort ;
diff --git a/mod-ams.lv2/vcaexp.ttl b/mod-ams.lv2/vcaexp.ttl
index ed74338..147d819 100644
--- a/mod-ams.lv2/vcaexp.ttl
+++ b/mod-ams.lv2/vcaexp.ttl
@@ -4,11 +4,6 @@
 @prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 @prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
 @prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/vcaexp/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vcaexp_gui.so> .
 
 <http://github.com/blablack/ams-lv2/vcaexp>
 	a lv2:Plugin, lv2:AmplifierPlugin ;
@@ -16,7 +11,6 @@
 	doap:name "AMS VCA Exp" ;
 	doap:license <http://opensource.org/licenses/isc> ;
 	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vcaexp/gui> ;
 	lv2:optionalFeature lv2:hardRTCapable ;
 	lv2:port [
 		a lv2:InputPort, lv2:CVPort ;
diff --git a/mod-ams.lv2/vcalin.ttl b/mod-ams.lv2/vcalin.ttl
index 24e9d44..bfc5445 100644
--- a/mod-ams.lv2/vcalin.ttl
+++ b/mod-ams.lv2/vcalin.ttl
@@ -4,11 +4,6 @@
 @prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 @prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
 @prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/vcalin/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vcalin_gui.so> .
 
 <http://github.com/blablack/ams-lv2/vcalin>
 	a lv2:Plugin, lv2:AmplifierPlugin ;
@@ -16,7 +11,6 @@
 	doap:name "AMS VCA Lin" ;
 	doap:license <http://opensource.org/licenses/isc> ;
 	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vcalin/gui> ;
 	lv2:optionalFeature lv2:hardRTCapable ;
 	lv2:port [
 		a lv2:InputPort, lv2:CVPort ;
diff --git a/mod-ams.lv2/vcf.ttl b/mod-ams.lv2/vcf.ttl
index 4aea079..3b298ea 100644
--- a/mod-ams.lv2/vcf.ttl
+++ b/mod-ams.lv2/vcf.ttl
@@ -4,11 +4,6 @@
 @prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 @prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
 @prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
-
-<http://github.com/blablack/ams-lv2/vcf/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vcf_gui.so> .
 
 <http://github.com/blablack/ams-lv2/vcf>
 	a lv2:Plugin, lv2:SpectralPlugin ;
@@ -16,7 +11,6 @@
 	doap:name "AMS VCF" ;
 	doap:license <http://opensource.org/licenses/isc> ;
 	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vcf/gui> ;
 	lv2:optionalFeature lv2:hardRTCapable ;
 	lv2:port 
 		[
diff --git a/mod-ams.lv2/vco3.ttl b/mod-ams.lv2/vco3.ttl
index 3c94b7c..2b30d55 100644
--- a/mod-ams.lv2/vco3.ttl
+++ b/mod-ams.lv2/vco3.ttl
@@ -3,15 +3,10 @@
 @prefix rdf: 	<http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
 @prefix rdfs: 	<http://www.w3.org/2000/01/rdf-schema#> .
 @prefix lv2: 	<http://lv2plug.in/ns/lv2core#> .
-@prefix guiext: <http://lv2plug.in/ns/extensions/ui#> .
 @prefix pg:     <http://lv2plug.in/ns/ext/port-groups#> .
 @prefix ll: 	<http://ll-plugins.nongnu.org/lv2/namespace#> .
 @prefix param:  <http://lv2plug.in/ns/ext/parameters#> .
 
-<http://github.com/blablack/ams-lv2/vco3/gui>
-  a guiext:GtkUI ;
-  guiext:binary <vco3_gui.so> .
-
 <http://github.com/blablack/ams-lv2/vco3#group_frequency>
     a param:OscillatorControls ;
     lv2:name "Frequency" ;
@@ -39,7 +34,6 @@
 	doap:name "AMS VCO3" ;
 	doap:license <http://opensource.org/licenses/isc> ;
 	ll:pegName "p" ;
-	guiext:ui <http://github.com/blablack/ams-lv2/vco3/gui> ;
 	lv2:optionalFeature lv2:hardRTCapable ;
 	lv2:port [
 		a lv2:CVPort, lv2:InputPort ;
