diff --git a/src/state.c b/src/state.c
index 9bc1d6d..6a2612a 100644
--- a/src/state.c
+++ b/src/state.c
@@ -942,6 +942,8 @@ lilv_state_write(LilvWorld*       world,
 	// Write property values with precise types
 	sratom_set_pretty_numbers(sratom, false);
 
+	if (getenv("LILV_STATE_SKIP_PROPERTIES") == NULL) {
+
 	// Write properties
 	const SerdNode state_node = serd_node_from_string(SERD_BLANK,
 	                                                  USTR("2state"));
@@ -970,6 +972,8 @@ lilv_state_write(LilvWorld*       world,
 		serd_writer_end_anon(writer, &state_node);
 	}
 
+	} // getenv("LILV_STATE_SKIP_PROPERTIES")
+
 	sratom_free(sratom);
 	return 0;
 }
