diff --git a/wscript b/wscript
index 787ee52..17f01a3 100644
--- a/wscript
+++ b/wscript
@@ -29,6 +29,8 @@ def configure(conf):
     else:
         conf.env.append_unique('CXXFLAGS', ['-O2','-funroll-loops','-std=c++0x'])
 
+    conf.env.append_unique('CXXFLAGS', ['-O3', '-fPIC','-fpermissive','-finline-functions'])
+
     if not autowaf.is_child():
         autowaf.check_pkg(conf, 'lv2core', uselib_store='LV2CORE')
 
