diff --git a/Plugin/CMakeLists.txt b/Plugin/CMakeLists.txt
--- a/Plugin/CMakeLists.txt
+++ b/Plugin/CMakeLists.txt
@@ -54,8 +54,8 @@ target_compile_definitions(CHOWTapeModel
     JUCE_WEB_BROWSER=0
     JUCE_USE_CURL=0
     JUCE_VST3_CAN_REPLACE_VST2=0
-    JUCE_JACK=1
-    JUCE_ALSA=1
+    JUCE_JACK=0
+    JUCE_ALSA=0
     CHOWDSP_AUTO_UPDATE=1
 )
 
diff --git a/Plugin/modules/CMakeLists.txt b/Plugin/modules/CMakeLists.txt
--- a/Plugin/modules/CMakeLists.txt
+++ b/Plugin/modules/CMakeLists.txt
@@ -34,6 +32,8 @@ target_compile_definitions(juce_plugin_modules
         JUCE_WEB_BROWSER=0
         JUCE_USE_CURL=0
         JUCE_VST3_CAN_REPLACE_VST2=0
+        JUCE_JACK=0
+        JUCE_ALSA=0
         FOLEYS_SHOW_GUI_EDITOR_PALLETTE=0
         FOLEYS_ENABLE_BINARY_DATA=1
         JucePlugin_Manufacturer="chowdsp"
diff --git a/Plugin/modules/JUCE/modules/juce_audio_devices/juce_audio_devices.h b/Plugin/modules/JUCE/modules/juce_audio_devices/juce_audio_devices.h
index e27163c..6577ffd 100644
--- a/Plugin/modules/JUCE/modules/juce_audio_devices/juce_audio_devices.h
+++ b/Plugin/modules/JUCE/modules/juce_audio_devices/juce_audio_devices.h
@@ -41,7 +41,7 @@
   dependencies:       juce_audio_basics, juce_events
   OSXFrameworks:      CoreAudio CoreMIDI AudioToolbox
   iOSFrameworks:      CoreAudio CoreMIDI AudioToolbox AVFoundation
-  linuxPackages:      alsa
+  linuxPackages:
   mingwLibs:          winmm
 
  END_JUCE_MODULE_DECLARATION
