diff --git a/resources/neural_amp_modeler.ttl.in b/resources/neural_amp_modeler.ttl.in
index 1125b8d..b8dd42e 100644
--- a/resources/neural_amp_modeler.ttl.in
+++ b/resources/neural_amp_modeler.ttl.in
@@ -106,7 +106,7 @@ A large collection of models is available at https://www.tone3000.com
 		lv2:index 6;
 		lv2:symbol "quality_scale";
 		lv2:name "Quality";
-		lv2:default 1.0;
+		lv2:default 0.0;
 		lv2:minimum 0.0;
-		lv2:maximum 1.0;
+		lv2:maximum 1.0;
 	].
